From e48f5b4073b9c1551dbea4ef5b15933672de3a1e Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Mon, 10 Jan 2011 14:43:00 -0600
Subject: [PATCH 8/9] Backport a lot of GI transfer annotations

This dramatically improves using GTK2 with g-i.

Taken from master commit e0aa12eb0ab8d without the non-documentation changes.
---
 gtk/gtkaccellabel.c      |    2 +-
 gtk/gtkaction.c          |   10 ++++----
 gtk/gtkactivatable.c     |    2 +-
 gtk/gtkassistant.c       |   23 ++++++++++---------
 gtk/gtkbuildable.c       |    6 ++--
 gtk/gtkbutton.c          |    2 +-
 gtk/gtkcellrenderer.c    |    8 +++---
 gtk/gtkcellview.c        |    2 +-
 gtk/gtkcolorseldialog.c  |    2 +-
 gtk/gtkcombobox.c        |    8 ++++--
 gtk/gtkcontainer.c       |    2 +-
 gtk/gtkdnd.c             |   19 +++++++--------
 gtk/gtkentry.c           |   16 +++++++-----
 gtk/gtkentrycompletion.c |    7 +++--
 gtk/gtkexpander.c        |    5 ++-
 gtk/gtkfilechooser.c     |   42 +++++++++++++++++-----------------
 gtk/gtkfontsel.c         |   56 +++++++++++++++++++++++----------------------
 gtk/gtkframe.c           |    2 +-
 gtk/gtkiconfactory.c     |    4 +-
 gtk/gtkicontheme.c       |   17 +++++++------
 gtk/gtkiconview.c        |    9 ++++---
 gtk/gtkimagemenuitem.c   |    5 ++-
 gtk/gtkinfobar.c         |    6 ++--
 gtk/gtkinvisible.c       |    2 +-
 gtk/gtklabel.c           |    4 +-
 gtk/gtklayout.c          |   26 ++++++++++----------
 gtk/gtkmain.c            |   11 ++++++++-
 gtk/gtkmenuitem.c        |    6 ++--
 gtk/gtkmenutoolbutton.c  |    3 +-
 gtk/gtkmountoperation.c  |    6 ++--
 gtk/gtknotebook.c        |   25 ++++++++++----------
 gtk/gtkoffscreenwindow.c |    5 ++-
 gtk/gtkpagesetup.c       |    6 ++--
 gtk/gtkpaned.c           |    6 ++--
 gtk/gtkplug.c            |    4 +-
 gtk/gtkprintcontext.c    |   12 +++++-----
 gtk/gtkprintoperation.c  |   18 +++++++-------
 gtk/gtkprintsettings.c   |   18 +++++++-------
 gtk/gtkradiobutton.c     |    6 +++-
 gtk/gtkradiomenuitem.c   |   27 ++++++++++++++-------
 gtk/gtkradiotoolbutton.c |   26 ++++++++++----------
 gtk/gtkrange.c           |    4 +-
 gtk/gtkrc.c              |   36 ++++++++++++++--------------
 gtk/gtkrecentchooser.c   |    6 ++--
 gtk/gtkrecentmanager.c   |   14 ++++++-----
 gtk/gtkscale.c           |   12 +++++-----
 gtk/gtkscalebutton.c     |    8 +++---
 gtk/gtkscrolledwindow.c  |   17 ++++++-------
 gtk/gtkselection.c       |    4 +-
 gtk/gtksettings.c        |   10 ++++----
 gtk/gtksocket.c          |    2 +-
 gtk/gtkspinbutton.c      |    2 +-
 gtk/gtkstatusbar.c       |    2 +-
 gtk/gtkstatusicon.c      |   11 +++++----
 gtk/gtkstyle.c           |   14 ++++++-----
 gtk/gtktextbuffer.c      |   15 +++++------
 gtk/gtktextiter.c        |   10 ++++----
 gtk/gtktexttagtable.c    |    2 +-
 gtk/gtktoolbar.c         |    8 +++---
 gtk/gtktoolbutton.c      |   26 +++++++++++----------
 gtk/gtktoolitem.c        |   28 ++++++++++++----------
 gtk/gtktoolitemgroup.c   |    6 ++--
 gtk/gtktreemodel.c       |   10 ++++----
 gtk/gtktreemodelsort.c   |    6 ++--
 gtk/gtktreeview.c        |   39 ++++++++++++++++---------------
 gtk/gtktreeviewcolumn.c  |   19 ++++++++-------
 gtk/gtkviewport.c        |    8 +++---
 gtk/gtkwidget.c          |   17 +++++++------
 gtk/gtkwindow.c          |    4 +-
 69 files changed, 425 insertions(+), 381 deletions(-)

diff --git a/gtk/gtkaccellabel.c b/gtk/gtkaccellabel.c
index da982bd..e0482b1 100644
--- a/gtk/gtkaccellabel.c
+++ b/gtk/gtkaccellabel.c
@@ -296,7 +296,7 @@ gtk_accel_label_finalize (GObject *object)
  * Fetches the widget monitored by this accelerator label. See
  * gtk_accel_label_set_accel_widget().
  *
- * Returns: the object monitored by the accelerator label, or %NULL.
+ * Returns: (transfer none): the object monitored by the accelerator label, or %NULL.
  **/
 GtkWidget*
 gtk_accel_label_get_accel_widget (GtkAccelLabel *accel_label)
diff --git a/gtk/gtkaction.c b/gtk/gtkaction.c
index 1cb1cec..8605ed4 100644
--- a/gtk/gtkaction.c
+++ b/gtk/gtkaction.c
@@ -869,7 +869,7 @@ gtk_action_unblock_activate (GtkAction *action)
  * This function is intended for use by action implementations to
  * create icons displayed in the proxy widgets.
  *
- * Returns: a widget that displays the icon for this action.
+ * Returns: (transfer full): a widget that displays the icon for this action.
  *
  * Since: 2.4
  */
@@ -895,7 +895,7 @@ gtk_action_create_icon (GtkAction *action, GtkIconSize icon_size)
  *
  * Creates a menu item widget that proxies for the given action.
  *
- * Returns: a menu item connected to the action.
+ * Returns: (transfer full): a menu item connected to the action.
  *
  * Since: 2.4
  */
@@ -920,7 +920,7 @@ gtk_action_create_menu_item (GtkAction *action)
  *
  * Creates a toolbar item widget that proxies for the given action.
  *
- * Returns: a toolbar item connected to the action.
+ * Returns: (transfer full): a toolbar item connected to the action.
  *
  * Since: 2.4
  */
@@ -1683,7 +1683,7 @@ gtk_action_set_gicon (GtkAction *action,
  *
  * Gets the gicon of @action.
  *
- * Returns: The action's #GIcon if one is set.
+ * Returns: (transfer none): The action's #GIcon if one is set.
  *
  * Since: 2.16
  */
@@ -1941,7 +1941,7 @@ gtk_action_disconnect_accelerator (GtkAction *action)
  * item or the toolbar item it creates, this function returns an
  * instance of that menu.
  *
- * Return value: the menu item provided by the action, or %NULL.
+ * Return value: (transfer full): the menu item provided by the action, or %NULL.
  *
  * Since: 2.12
  */
diff --git a/gtk/gtkactivatable.c b/gtk/gtkactivatable.c
index 85abdec..13ce3dd 100644
--- a/gtk/gtkactivatable.c
+++ b/gtk/gtkactivatable.c
@@ -496,7 +496,7 @@ gtk_activatable_do_set_related_action (GtkActivatable *activatable,
  *
  * Gets the related #GtkAction for @activatable.
  *
- * Returns: the related #GtkAction if one is set.
+ * Returns: (transfer none): the related #GtkAction if one is set.
  *
  * Since: 2.16
  **/
diff --git a/gtk/gtkassistant.c b/gtk/gtkassistant.c
index 8f18f13..72314aa 100644
--- a/gtk/gtkassistant.c
+++ b/gtk/gtkassistant.c
@@ -1645,7 +1645,8 @@ gtk_assistant_get_n_pages (GtkAssistant *assistant)
  *
  * Returns the child widget contained in page number @page_num.
  *
- * Return value: The child widget, or %NULL if @page_num is out of bounds.
+ * Return value: (transfer none): The child widget, or %NULL
+ *     if @page_num is out of bounds.
  *
  * Since: 2.10
  **/
@@ -2076,11 +2077,11 @@ gtk_assistant_set_page_header_image (GtkAssistant *assistant,
  * gtk_assistant_get_page_header_image:
  * @assistant: a #GtkAssistant
  * @page: a page of @assistant
- * 
- * Gets the header image for @page. 
- * 
- * Return value: the header image for @page, or %NULL
- * if there's no header image for the page.
+ *
+ * Gets the header image for @page.
+ *
+ * Return value: (transfer none): the header image for @page, or %NULL
+ *     if there's no header image for the page.
  *
  * Since: 2.10
  **/
@@ -2156,11 +2157,11 @@ gtk_assistant_set_page_side_image (GtkAssistant *assistant,
  * gtk_assistant_get_page_side_image:
  * @assistant: a #GtkAssistant
  * @page: a page of @assistant
- * 
- * Gets the header image for @page. 
- * 
- * Return value: the side image for @page, or %NULL
- * if there's no side image for the page.
+ *
+ * Gets the header image for @page.
+ *
+ * Return value: (transfer none): the side image for @page, or %NULL
+ *     if there's no side image for the page.
  *
  * Since: 2.10
  **/
diff --git a/gtk/gtkbuildable.c b/gtk/gtkbuildable.c
index 68e6f8f..6605305 100644
--- a/gtk/gtkbuildable.c
+++ b/gtk/gtkbuildable.c
@@ -203,12 +203,12 @@ gtk_buildable_parser_finished (GtkBuildable *buildable,
  * @builder: #GtkBuilder used to construct this object
  * @name: name of child to construct
  *
- * Constructs a child of @buildable with the name @name. 
+ * Constructs a child of @buildable with the name @name.
  *
  * #GtkBuilder calls this function if a "constructor" has been
  * specified in the UI definition.
  *
- * Returns: the constructed child
+ * Returns: (transfer full): the constructed child
  *
  * Since: 2.12
  **/
@@ -336,7 +336,7 @@ gtk_buildable_custom_finished (GtkBuildable  *buildable,
  *
  * Get the internal child called @childname of the @buildable object.
  *
- * Returns: the internal child of the buildable object 
+ * Returns: (transfer none): the internal child of the buildable object
  *
  * Since: 2.12
  **/
diff --git a/gtk/gtkbutton.c b/gtk/gtkbutton.c
index 80f555e..8292c2c 100644
--- a/gtk/gtkbutton.c
+++ b/gtk/gtkbutton.c
@@ -2250,7 +2250,7 @@ gtk_button_set_image (GtkButton *button,
  * This may have been explicitly set by gtk_button_set_image()
  * or constructed by gtk_button_new_from_stock().
  *
- * Return value: a #GtkWidget or %NULL in case there is no image
+ * Return value: (transfer none): a #GtkWidget or %NULL in case there is no image
  *
  * Since: 2.6
  */
diff --git a/gtk/gtkcellrenderer.c b/gtk/gtkcellrenderer.c
index b55b098..7c169c1 100644
--- a/gtk/gtkcellrenderer.c
+++ b/gtk/gtkcellrenderer.c
@@ -632,15 +632,15 @@ gtk_cell_renderer_activate (GtkCellRenderer      *cell,
  * @cell: a #GtkCellRenderer
  * @event: a #GdkEvent
  * @widget: widget that received the event
- * @path: widget-dependent string representation of the event location; 
+ * @path: widget-dependent string representation of the event location;
  *    e.g. for #GtkTreeView, a string representation of #GtkTreePath
  * @background_area: background area as passed to gtk_cell_renderer_render()
  * @cell_area: cell area as passed to gtk_cell_renderer_render()
  * @flags: render flags
- * 
+ *
  * Passes an activate event to the cell renderer for possible processing.
- * 
- * Return value: A new #GtkCellEditable, or %NULL
+ *
+ * Return value: (transfer full): A new #GtkCellEditable, or %NULL
  **/
 GtkCellEditable *
 gtk_cell_renderer_start_editing (GtkCellRenderer      *cell,
diff --git a/gtk/gtkcellview.c b/gtk/gtkcellview.c
index 0dd98b0..0dcfdad 100644
--- a/gtk/gtkcellview.c
+++ b/gtk/gtkcellview.c
@@ -895,7 +895,7 @@ gtk_cell_view_set_model (GtkCellView  *cell_view,
  * Returns the model for @cell_view. If no model is used %NULL is
  * returned.
  *
- * Returns: a #GtkTreeModel used or %NULL
+ * Returns: (transfer none): a #GtkTreeModel used or %NULL
  *
  * Since: 2.16
  **/
diff --git a/gtk/gtkcolorseldialog.c b/gtk/gtkcolorseldialog.c
index 977cb15..8ec2d1b 100644
--- a/gtk/gtkcolorseldialog.c
+++ b/gtk/gtkcolorseldialog.c
@@ -193,7 +193,7 @@ gtk_color_selection_dialog_new (const gchar *title)
  *
  * Retrieves the #GtkColorSelection widget embedded in the dialog.
  *
- * Returns: the embedded #GtkColorSelection
+ * Returns: (transfer none): the embedded #GtkColorSelection
  *
  * Since: 2.14
  **/
diff --git a/gtk/gtkcombobox.c b/gtk/gtkcombobox.c
index acafa1a..de42487 100644
--- a/gtk/gtkcombobox.c
+++ b/gtk/gtkcombobox.c
@@ -5294,12 +5294,13 @@ out:
 }
 
 /**
- * gtk_combo_box_get_model
+ * gtk_combo_box_get_model:
  * @combo_box: A #GtkComboBox
  *
  * Returns the #GtkTreeModel which is acting as data source for @combo_box.
  *
- * Return value: (transfer none): A #GtkTreeModel which was passed during construction.
+ * Return value: (transfer none): A #GtkTreeModel which was passed
+ *     during construction.
  *
  * Since: 2.4
  */
@@ -6074,7 +6075,8 @@ gtk_combo_box_set_title (GtkComboBox *combo_box,
  * This function is mostly intended for use by accessibility technologies;
  * applications should have little use for it.
  *
- * Returns: the accessible object corresponding to the combo box's popup.
+ * Returns: (transfer none): the accessible object corresponding
+ *     to the combo box's popup.
  *
  * Since: 2.6
  */
diff --git a/gtk/gtkcontainer.c b/gtk/gtkcontainer.c
index e801e5e..16402d9 100644
--- a/gtk/gtkcontainer.c
+++ b/gtk/gtkcontainer.c
@@ -1509,7 +1509,7 @@ gtk_container_forall (GtkContainer *container,
 /**
  * gtk_container_foreach:
  * @container: a #GtkContainer
- * @callback: a callback
+ * @callback: (scope call):  a callback
  * @callback_data: callback user data
  * 
  * Invokes @callback on each non-internal child of @container. See
diff --git a/gtk/gtkdnd.c b/gtk/gtkdnd.c
index c378668..8571042 100644
--- a/gtk/gtkdnd.c
+++ b/gtk/gtkdnd.c
@@ -990,17 +990,16 @@ gtk_drag_get_data (GtkWidget      *widget,
 }
 
 
-/*************************************************************
+/**
  * gtk_drag_get_source_widget:
- *     Get the widget the was the source of this drag, if
- *     the drag originated from this application.
- *   arguments:
- *     context: The drag context for this drag
- *   results:
- *     The source widget, or NULL if the drag originated from
- *     a different application.
- *************************************************************/
-
+ * @context: a (destination side) drag context
+ *
+ * Determines the source widget for a drag.
+ *
+ * Return value: (transfer none): if the drag is occurring
+ *     within a single application, a pointer to the source widget.
+ *     Otherwise, %NULL.
+ */
 GtkWidget *
 gtk_drag_get_source_widget (GdkDragContext *context)
 {
diff --git a/gtk/gtkentry.c b/gtk/gtkentry.c
index 674a6fb..dd509c2 100644
--- a/gtk/gtkentry.c
+++ b/gtk/gtkentry.c
@@ -6685,7 +6685,7 @@ get_buffer (GtkEntry *entry)
  *
  * Since: 2.18
  *
- * Returns: A #GtkEntryBuffer object.
+ * Returns: (transfer none): A #GtkEntryBuffer object.
  */
 GtkEntryBuffer*
 gtk_entry_get_buffer (GtkEntry *entry)
@@ -6770,7 +6770,7 @@ gtk_entry_set_buffer (GtkEntry       *entry,
  *
  * See also gtk_entry_get_icon_window().
  *
- * Return value: the entry's text window.
+ * Return value: (transfer none): the entry's text window.
  *
  * Since: 2.20
  **/
@@ -7944,7 +7944,8 @@ gtk_entry_get_icon_activatable (GtkEntry             *entry,
  * method will work regardless of whether the icon was set using a
  * #GdkPixbuf, a #GIcon, a stock item, or an icon name.
  *
- * Returns: A #GdkPixbuf, or %NULL if no icon is set for this position.
+ * Returns: (transfer none): A #GdkPixbuf, or %NULL if no icon is
+ *     set for this position.
  *
  * Since: 2.16
  */
@@ -7978,8 +7979,8 @@ gtk_entry_get_icon_pixbuf (GtkEntry             *entry,
  * no icon or if the icon was set by some other method (e.g., by
  * stock, pixbuf, or icon name).
  *
- * Returns: A #GIcon, or %NULL if no icon is set or if the icon
- *          is not a #GIcon
+ * Returns: (transfer none): A #GIcon, or %NULL if no icon is set
+ *     or if the icon is not a #GIcon
  *
  * Since: 2.16
  */
@@ -8309,7 +8310,7 @@ gtk_entry_get_current_icon_drag_source (GtkEntry *entry)
  *
  * See also gtk_entry_get_text_window().
  *
- * Return value: the entry's icon window at @icon_pos.
+ * Return value: (transfer none): the entry's icon window at @icon_pos.
  *
  * Since: 2.20
  */
@@ -9833,7 +9834,8 @@ gtk_entry_set_completion (GtkEntry           *entry,
  *
  * Returns the auxiliary completion object currently in use by @entry.
  *
- * Return value: The auxiliary completion object currently in use by @entry.
+ * Return value: (transfer none): The auxiliary completion object currently
+ *     in use by @entry.
  *
  * Since: 2.4
  */
diff --git a/gtk/gtkentrycompletion.c b/gtk/gtkentrycompletion.c
index 68b758a..05b817c 100644
--- a/gtk/gtkentrycompletion.c
+++ b/gtk/gtkentrycompletion.c
@@ -998,7 +998,7 @@ gtk_entry_completion_new (void)
  *
  * Gets the entry @completion has been attached to.
  *
- * Return value: The entry @completion has been attached to.
+ * Return value: (transfer none): The entry @completion has been attached to.
  *
  * Since: 2.4
  */
@@ -1062,7 +1062,8 @@ gtk_entry_completion_set_model (GtkEntryCompletion *completion,
  * Returns the model the #GtkEntryCompletion is using as data source.
  * Returns %NULL if the model is unset.
  *
- * Return value: A #GtkTreeModel, or %NULL if none is currently being used.
+ * Return value: (transfer none): A #GtkTreeModel, or %NULL if none
+ *     is currently being used.
  *
  * Since: 2.4
  */
@@ -1073,7 +1074,7 @@ gtk_entry_completion_get_model (GtkEntryCompletion *completion)
 
   if (!completion->priv->filter_model)
     return NULL;
-  
+
   return gtk_tree_model_filter_get_model (completion->priv->filter_model);
 }
 
diff --git a/gtk/gtkexpander.c b/gtk/gtkexpander.c
index b69729f..693c867 100644
--- a/gtk/gtkexpander.c
+++ b/gtk/gtkexpander.c
@@ -1758,8 +1758,9 @@ gtk_expander_set_label_widget (GtkExpander *expander,
  * Retrieves the label widget for the frame. See
  * gtk_expander_set_label_widget().
  *
- * Return value: the label widget, or %NULL if there is none.
- * 
+ * Return value: (transfer none): the label widget,
+ *     or %NULL if there is none.
+ *
  * Since: 2.4
  **/
 GtkWidget *
diff --git a/gtk/gtkfilechooser.c b/gtk/gtkfilechooser.c
index 85f972f..a8cb1a8 100644
--- a/gtk/gtkfilechooser.c
+++ b/gtk/gtkfilechooser.c
@@ -1586,7 +1586,7 @@ gtk_file_chooser_set_current_folder_uri (GtkFileChooser *chooser,
 /**
  * gtk_file_chooser_get_current_folder_uri:
  * @chooser: a #GtkFileChooser
- * 
+ *
  * Gets the current folder of @chooser as an URI.
  * See gtk_file_chooser_set_current_folder_uri().
  *
@@ -1628,7 +1628,7 @@ gtk_file_chooser_get_current_folder_uri (GtkFileChooser *chooser)
  * @chooser: a #GtkFileChooser
  * @file: the #GFile for the new folder
  * @error: (allow-none): location to store error, or %NULL.
- * 
+ *
  * Sets the current folder for @chooser from a #GFile.
  * Internal function, see gtk_file_chooser_set_current_folder_uri().
  *
@@ -1652,11 +1652,11 @@ gtk_file_chooser_set_current_folder_file (GtkFileChooser  *chooser,
 /**
  * gtk_file_chooser_get_current_folder_file:
  * @chooser: a #GtkFileChooser
- * 
+ *
  * Gets the current folder of @chooser as #GFile.
  * See gtk_file_chooser_get_current_folder_uri().
- * 
- * Return value: the #GFile for the current folder.
+ *
+ * Return value: (transfer full): the #GFile for the current folder.
  *
  * Since: 2.14
  */
@@ -1665,7 +1665,7 @@ gtk_file_chooser_get_current_folder_file (GtkFileChooser *chooser)
 {
   g_return_val_if_fail (GTK_IS_FILE_CHOOSER (chooser), NULL);
 
-  return GTK_FILE_CHOOSER_GET_IFACE (chooser)->get_current_folder (chooser);  
+  return GTK_FILE_CHOOSER_GET_IFACE (chooser)->get_current_folder (chooser);
 }
 
 /**
@@ -1804,8 +1804,8 @@ gtk_file_chooser_set_file (GtkFileChooser  *chooser,
  * If the file chooser is in folder mode, this function returns the selected
  * folder.
  *
- * Returns: a selected #GFile. You own the returned file; use
- *          g_object_unref() to release it.
+ * Returns: (transfer full): a selected #GFile. You own the returned file;
+ *     use g_object_unref() to release it.
  *
  * Since: 2.14
  **/
@@ -1885,11 +1885,11 @@ gtk_file_chooser_set_preview_widget (GtkFileChooser *chooser,
 /**
  * gtk_file_chooser_get_preview_widget:
  * @chooser: a #GtkFileChooser
- * 
+ *
  * Gets the current preview widget; see
  * gtk_file_chooser_set_preview_widget().
- * 
- * Return value: the current preview widget, or %NULL
+ *
+ * Return value: (transfer none): the current preview widget, or %NULL
  *
  * Since: 2.4
  **/
@@ -2007,12 +2007,12 @@ gtk_file_chooser_get_use_preview_label (GtkFileChooser *chooser)
 /**
  * gtk_file_chooser_get_preview_file:
  * @chooser: a #GtkFileChooser
- * 
+ *
  * Gets the #GFile that should be previewed in a custom preview
  * Internal function, see gtk_file_chooser_get_preview_uri().
- * 
- * Return value: the #GFile for the file to preview, or %NULL if no file
- *  is selected. Free with g_object_unref().
+ *
+ * Return value: (transfer none): the #GFile for the file to preview,
+ *     or %NULL if no file is selected. Free with g_object_unref().
  *
  * Since: 2.14
  **/
@@ -2156,11 +2156,11 @@ gtk_file_chooser_set_extra_widget (GtkFileChooser *chooser,
 /**
  * gtk_file_chooser_get_extra_widget:
  * @chooser: a #GtkFileChooser
- * 
+ *
  * Gets the current preview widget; see
  * gtk_file_chooser_set_extra_widget().
- * 
- * Return value: the current extra widget, or %NULL
+ *
+ * Return value: (transfer none): the current extra widget, or %NULL
  *
  * Since: 2.4
  **/
@@ -2273,10 +2273,10 @@ gtk_file_chooser_set_filter (GtkFileChooser *chooser,
 /**
  * gtk_file_chooser_get_filter:
  * @chooser: a #GtkFileChooser
- * 
+ *
  * Gets the current filter; see gtk_file_chooser_set_filter().
- * 
- * Return value: the current filter, or %NULL
+ *
+ * Return value: (transfer none): the current filter, or %NULL
  *
  * Since: 2.4
  **/
diff --git a/gtk/gtkfontsel.c b/gtk/gtkfontsel.c
index eeeb4ac..0ce63fa 100644
--- a/gtk/gtkfontsel.c
+++ b/gtk/gtkfontsel.c
@@ -1191,8 +1191,8 @@ gtk_font_selection_get_font_internal (GtkFontSelection *fontsel)
  *
  * This returns the #GtkTreeView that lists font families, for
  * example, 'Sans', 'Serif', etc.
- * 
- * Return value: A #GtkWidget that is part of @fontsel
+ *
+ * Return value: (transfer none): A #GtkWidget that is part of @fontsel
  *
  * Since: 2.14
  */
@@ -1211,7 +1211,7 @@ gtk_font_selection_get_family_list (GtkFontSelection *fontsel)
  * This returns the #GtkTreeView which lists all styles available for
  * the selected font. For example, 'Regular', 'Bold', etc.
  * 
- * Return value: A #GtkWidget that is part of @fontsel
+ * Return value: (transfer none): A #GtkWidget that is part of @fontsel
  *
  * Since: 2.14
  */
@@ -1228,9 +1228,9 @@ gtk_font_selection_get_face_list (GtkFontSelection *fontsel)
  * @fontsel: a #GtkFontSelection
  *
  * This returns the #GtkEntry used to allow the user to edit the font
- * number manually instead of selecting it from the list of font sizes. 
- * 
- * Return value: A #GtkWidget that is part of @fontsel
+ * number manually instead of selecting it from the list of font sizes.
+ *
+ * Return value: (transfer none): A #GtkWidget that is part of @fontsel
  *
  * Since: 2.14
  */
@@ -1246,9 +1246,9 @@ gtk_font_selection_get_size_entry (GtkFontSelection *fontsel)
  * gtk_font_selection_get_size_list:
  * @fontsel: a #GtkFontSelection
  *
- * This returns the #GtkTreeeView used to list font sizes. 
- * 
- * Return value: A #GtkWidget that is part of @fontsel
+ * This returns the #GtkTreeeView used to list font sizes.
+ *
+ * Return value: (transfer none): A #GtkWidget that is part of @fontsel
  *
  * Since: 2.14
  */
@@ -1263,10 +1263,10 @@ gtk_font_selection_get_size_list (GtkFontSelection *fontsel)
 /**
  * gtk_font_selection_get_preview_entry:
  * @fontsel: a #GtkFontSelection
- * 
+ *
  * This returns the #GtkEntry used to display the font as a preview.
  *
- * Return value: A #GtkWidget that is part of @fontsel
+ * Return value: (transfer none): A #GtkWidget that is part of @fontsel
  *
  * Since: 2.14
  */
@@ -1281,13 +1281,13 @@ gtk_font_selection_get_preview_entry (GtkFontSelection *fontsel)
 /**
  * gtk_font_selection_get_family:
  * @fontsel: a #GtkFontSelection
- * 
+ *
  * Gets the #PangoFontFamily representing the selected font family.
  *
- * Return value: A #PangoFontFamily representing the selected font
- *     family. Font families are a collection of font faces. The 
- *     returned object is owned by @fontsel and must not be modified 
- *     or freed.
+ * Return value: (transfer none): A #PangoFontFamily representing the
+ *     selected font family. Font families are a collection of font
+ *     faces. The returned object is owned by @fontsel and must not
+ *     be modified or freed.
  *
  * Since: 2.14
  */
@@ -1302,13 +1302,13 @@ gtk_font_selection_get_family (GtkFontSelection *fontsel)
 /**
  * gtk_font_selection_get_face:
  * @fontsel: a #GtkFontSelection
- * 
+ *
  * Gets the #PangoFontFace representing the selected font group
- * details (i.e. family, slant, weight, width, etc).   
+ * details (i.e. family, slant, weight, width, etc).
  *
- * Return value: A #PangoFontFace representing the selected font 
- *     group details. The returned object is owned by @fontsel and
- *     must not be modified or freed. 
+ * Return value: (transfer none): A #PangoFontFace representing the
+ *     selected font group details. The returned object is owned by
+ *     @fontsel and must not be modified or freed.
  *
  * Since: 2.14
  */
@@ -1323,10 +1323,10 @@ gtk_font_selection_get_face (GtkFontSelection *fontsel)
 /**
  * gtk_font_selection_get_size:
  * @fontsel: a #GtkFontSelection
- * 
+ *
  * The selected font size.
  *
- * Return value: A n integer representing the selected font size, 
+ * Return value: A n integer representing the selected font size,
  *     or -1 if no font size is selected.
  *
  * Since: 2.14
@@ -1676,11 +1676,11 @@ gtk_font_selection_dialog_new (const gchar *title)
 
 /**
  * gtk_font_selection_dialog_get_font_selection:
- * @colorsel: a #GtkFontSelectionDialog
+ * @fsd: a #GtkFontSelectionDialog
  *
  * Retrieves the #GtkFontSelection widget embedded in the dialog.
  *
- * Returns: the embedded #GtkFontSelection
+ * Returns: (transfer none): the embedded #GtkFontSelection
  *
  * Since: 2.22
  **/
@@ -1698,7 +1698,8 @@ gtk_font_selection_dialog_get_font_selection (GtkFontSelectionDialog *fsd)
  *
  * Gets the 'OK' button.
  *
- * Return value: the #GtkWidget used in the dialog for the 'OK' button.
+ * Return value: (transfer none): the #GtkWidget used in the dialog
+ *     for the 'OK' button.
  *
  * Since: 2.14
  */
@@ -1736,7 +1737,8 @@ gtk_font_selection_dialog_get_apply_button (GtkFontSelectionDialog *fsd)
  *
  * Gets the 'Cancel' button.
  *
- * Return value: the #GtkWidget used in the dialog for the 'Cancel' button.
+ * Return value: (transfer none): the #GtkWidget used in the dialog
+ *     for the 'Cancel' button.
  *
  * Since: 2.14
  */
diff --git a/gtk/gtkframe.c b/gtk/gtkframe.c
index d3fb6e3..8f2f9fe 100644
--- a/gtk/gtkframe.c
+++ b/gtk/gtkframe.c
@@ -408,7 +408,7 @@ gtk_frame_set_label_widget (GtkFrame  *frame,
  * Retrieves the label widget for the frame. See
  * gtk_frame_set_label_widget().
  *
- * Return value: the label widget, or %NULL if there is none.
+ * Return value: (transfer none): the label widget, or %NULL if there is none.
  **/
 GtkWidget *
 gtk_frame_get_label_widget (GtkFrame *frame)
diff --git a/gtk/gtkiconfactory.c b/gtk/gtkiconfactory.c
index 0b93b21..acfee7a 100644
--- a/gtk/gtkiconfactory.c
+++ b/gtk/gtkiconfactory.c
@@ -1619,7 +1619,7 @@ render_fallback_image (GtkStyle          *style,
  * (perhaps because an image file fails to load), a default "missing
  * image" icon will be returned instead.
  *
- * Return value: a #GdkPixbuf to be displayed
+ * Return value: (transfer full): a #GdkPixbuf to be displayed
  */
 GdkPixbuf*
 gtk_icon_set_render_icon (GtkIconSet        *icon_set,
@@ -2122,7 +2122,7 @@ gtk_icon_source_get_icon_name (const GtkIconSource *source)
  * virtual function. The reference count on the pixbuf is
  * not incremented.
  *
- * Return value: source pixbuf
+ * Return value: (transfer none): source pixbuf
  */
 GdkPixbuf*
 gtk_icon_source_get_pixbuf (const GtkIconSource *source)
diff --git a/gtk/gtkicontheme.c b/gtk/gtkicontheme.c
index 23a6a94..e42e3c8 100644
--- a/gtk/gtkicontheme.c
+++ b/gtk/gtkicontheme.c
@@ -3044,8 +3044,9 @@ icon_info_ensure_scale_and_pixbuf (GtkIconInfo  *icon_info,
 /**
  * gtk_icon_info_load_icon:
  * @icon_info: a #GtkIconInfo structure from gtk_icon_theme_lookup_icon()
- * @error: (allow-none): location to store error information on failure, or %NULL.
- * 
+ * @error: (allow-none): location to store error information on failure,
+ *     or %NULL.
+ *
  * Renders an icon previously looked up in an icon theme using
  * gtk_icon_theme_lookup_icon(); the size will be based on the size
  * passed to gtk_icon_theme_lookup_icon(). Note that the resulting
@@ -3056,12 +3057,12 @@ icon_info_ensure_scale_and_pixbuf (GtkIconInfo  *icon_info,
  * up too far. (This maintains sharpness.). This behaviour can be changed
  * by passing the %GTK_ICON_LOOKUP_FORCE_SIZE flag when obtaining
  * the #GtkIconInfo. If this flag has been specified, the pixbuf
- * returned by this function will be scaled to the exact size. 
- * 
- * Return value: the rendered icon; this may be a newly created icon
- *  or a new reference to an internal icon, so you must not modify
- *  the icon. Use g_object_unref() to release your reference to the
- *  icon.
+ * returned by this function will be scaled to the exact size.
+ *
+ * Return value: (transfer full): the rendered icon; this may be a newly
+ *     created icon or a new reference to an internal icon, so you must
+ *     not modify the icon. Use g_object_unref() to release your reference
+ *     to the icon.
  *
  * Since: 2.4
  **/
diff --git a/gtk/gtkiconview.c b/gtk/gtkiconview.c
index 59a1361..caf773d 100644
--- a/gtk/gtkiconview.c
+++ b/gtk/gtkiconview.c
@@ -5238,7 +5238,7 @@ gtk_icon_view_get_visible_range (GtkIconView  *icon_view,
 /**
  * gtk_icon_view_selected_foreach:
  * @icon_view: A #GtkIconView.
- * @func: The funcion to call for each selected icon.
+ * @func: (scope call): The function to call for each selected icon.
  * @data: User data to pass to the function.
  * 
  * Calls a function for each selected icon. Note that the model or
@@ -5439,7 +5439,8 @@ gtk_icon_view_set_model (GtkIconView *icon_view,
  * Returns the model the #GtkIconView is based on.  Returns %NULL if the
  * model is unset.
  *
- * Return value: A #GtkTreeModel, or %NULL if none is currently being used.
+ * Return value: (transfer none): A #GtkTreeModel, or %NULL if none is
+ *     currently being used.
  *
  * Since: 2.6 
  **/
@@ -7501,10 +7502,10 @@ gtk_icon_view_get_dest_item_at_pos (GtkIconView              *icon_view,
  * @icon_view: a #GtkIconView
  * @path: a #GtkTreePath in @icon_view
  *
- * Creates a #GdkPixmap representation of the item at @path.  
+ * Creates a #GdkPixmap representation of the item at @path.
  * This image is used for a drag icon.
  *
- * Return value: a newly-allocated pixmap of the drag icon.
+ * Return value: (transfer full): a newly-allocated pixmap of the drag icon.
  * 
  * Since: 2.8
  **/
diff --git a/gtk/gtkimagemenuitem.c b/gtk/gtkimagemenuitem.c
index f634d71..a214cf2 100644
--- a/gtk/gtkimagemenuitem.c
+++ b/gtk/gtkimagemenuitem.c
@@ -951,11 +951,12 @@ gtk_image_menu_item_set_image (GtkImageMenuItem *image_menu_item,
 
 /**
  * gtk_image_menu_item_get_image:
- * @image_menu_item: a #GtkImageMenuItem.
- * @returns: the widget set as image of @image_menu_item.
+ * @image_menu_item: a #GtkImageMenuItem
  *
  * Gets the widget that is currently set as the image of @image_menu_item.
  * See gtk_image_menu_item_set_image().
+ *
+ * Return value: (transfer none): the widget set as image of @image_menu_item
  **/
 GtkWidget*
 gtk_image_menu_item_get_image (GtkImageMenuItem *image_menu_item)
diff --git a/gtk/gtkinfobar.c b/gtk/gtkinfobar.c
index f49155b..66700ee 100644
--- a/gtk/gtkinfobar.c
+++ b/gtk/gtkinfobar.c
@@ -744,7 +744,7 @@ gtk_info_bar_add_action_widget (GtkInfoBar *info_bar,
  *
  * Returns the action area of @info_bar.
  *
- * Returns: the action area.
+ * Returns: (transfer none): the action area
  *
  * Since: 2.18
  */
@@ -762,7 +762,7 @@ gtk_info_bar_get_action_area (GtkInfoBar *info_bar)
  *
  * Returns the content area of @info_bar.
  *
- * Returns: the content area.
+ * Returns: (transfer none): the content area
  *
  * Since: 2.18
  */
@@ -786,7 +786,7 @@ gtk_info_bar_get_content_area (GtkInfoBar *info_bar)
  * to the end of the info bars's action area. The button widget is
  * returned, but usually you don't need it.
  *
- * Returns: the button widget that was added
+ * Returns: (transfer none): the button widget that was added
  *
  * Since: 2.18
  */
diff --git a/gtk/gtkinvisible.c b/gtk/gtkinvisible.c
index 8c0b3be..dbe2212 100644
--- a/gtk/gtkinvisible.c
+++ b/gtk/gtkinvisible.c
@@ -199,7 +199,7 @@ gtk_invisible_set_screen (GtkInvisible *invisible,
  *
  * Returns the #GdkScreen object associated with @invisible
  *
- * Return value: the associated #GdkScreen.
+ * Return value: (transfer none): the associated #GdkScreen.
  *
  * Since: 2.2
  **/
diff --git a/gtk/gtklabel.c b/gtk/gtklabel.c
index beba0b7..11e5912 100644
--- a/gtk/gtklabel.c
+++ b/gtk/gtklabel.c
@@ -1732,8 +1732,8 @@ gtk_label_set_mnemonic_widget (GtkLabel  *label,
  * Retrieves the target of the mnemonic (keyboard shortcut) of this
  * label. See gtk_label_set_mnemonic_widget().
  *
- * Return value: the target of the label's mnemonic, or %NULL if none
- *               has been set and the default algorithm will be used.
+ * Return value: (transfer none): the target of the label's mnemonic,
+ *     or %NULL if none has been set and the default algorithm will be used.
  **/
 GtkWidget *
 gtk_label_get_mnemonic_widget (GtkLabel *label)
diff --git a/gtk/gtklayout.c b/gtk/gtklayout.c
index 7e3e0b9..fc4d07d 100644
--- a/gtk/gtklayout.c
+++ b/gtk/gtklayout.c
@@ -145,10 +145,10 @@ gtk_layout_new (GtkAdjustment *hadjustment,
 /**
  * gtk_layout_get_bin_window:
  * @layout: a #GtkLayout
- * 
+ *
  * Retrieve the bin window of the layout used for drawing operations.
- * 
- * Return value: a #GdkWindow
+ *
+ * Return value: (transfer none): a #GdkWindow
  *
  * Since: 2.14
  **/
@@ -163,18 +163,18 @@ gtk_layout_get_bin_window (GtkLayout *layout)
 /**
  * gtk_layout_get_hadjustment:
  * @layout: a #GtkLayout
- * 
+ *
  * This function should only be called after the layout has been
  * placed in a #GtkScrolledWindow or otherwise configured for
  * scrolling. It returns the #GtkAdjustment used for communication
  * between the horizontal scrollbar and @layout.
  *
  * See #GtkScrolledWindow, #GtkScrollbar, #GtkAdjustment for details.
- * 
- * Return value: horizontal scroll adjustment
+ *
+ * Return value: (transfer none): horizontal scroll adjustment
  **/
-GtkAdjustment* 
-gtk_layout_get_hadjustment (GtkLayout     *layout)
+GtkAdjustment*
+gtk_layout_get_hadjustment (GtkLayout *layout)
 {
   g_return_val_if_fail (GTK_IS_LAYOUT (layout), NULL);
 
@@ -183,18 +183,18 @@ gtk_layout_get_hadjustment (GtkLayout     *layout)
 /**
  * gtk_layout_get_vadjustment:
  * @layout: a #GtkLayout
- * 
+ *
  * This function should only be called after the layout has been
  * placed in a #GtkScrolledWindow or otherwise configured for
  * scrolling. It returns the #GtkAdjustment used for communication
  * between the vertical scrollbar and @layout.
  *
  * See #GtkScrolledWindow, #GtkScrollbar, #GtkAdjustment for details.
- * 
- * Return value: vertical scroll adjustment
+ *
+ * Return value: (transfer none): vertical scroll adjustment
  **/
-GtkAdjustment* 
-gtk_layout_get_vadjustment (GtkLayout     *layout)
+GtkAdjustment*
+gtk_layout_get_vadjustment (GtkLayout *layout)
 {
   g_return_val_if_fail (GTK_IS_LAYOUT (layout), NULL);
 
diff --git a/gtk/gtkmain.c b/gtk/gtkmain.c
index f9c50e9..09a59b1 100644
--- a/gtk/gtkmain.c
+++ b/gtk/gtkmain.c
@@ -1869,6 +1869,14 @@ gtk_grab_add (GtkWidget *widget)
     }
 }
 
+/**
+ * gtk_grab_get_current:
+ *
+ * Queries the current grab of the default window group.
+ *
+ * Return value: (transfer none): The widget which currently
+ *     has the grab or %NULL if no grab is active
+ */
 GtkWidget*
 gtk_grab_get_current (void)
 {
@@ -2337,7 +2345,8 @@ gtk_get_current_event_state (GdkModifierType *state)
  * returns %NULL, otherwise returns the widget that received the event
  * originally.
  * 
- * Return value: the widget that originally received @event, or %NULL
+ * Return value: (transfer none): the widget that originally
+ *     received @event, or %NULL
  **/
 GtkWidget*
 gtk_get_event_widget (GdkEvent *event)
diff --git a/gtk/gtkmenuitem.c b/gtk/gtkmenuitem.c
index 8e03b99..cf1442a 100644
--- a/gtk/gtkmenuitem.c
+++ b/gtk/gtkmenuitem.c
@@ -799,10 +799,10 @@ gtk_menu_item_set_submenu (GtkMenuItem *menu_item,
  * gtk_menu_item_get_submenu:
  * @menu_item: a #GtkMenuItem
  *
- * Gets the submenu underneath this menu item, if any. See
- * gtk_menu_item_set_submenu().
+ * Gets the submenu underneath this menu item, if any.
+ * See gtk_menu_item_set_submenu().
  *
- * Return value: submenu for this menu item, or %NULL if none.
+ * Return value: (transfer none): submenu for this menu item, or %NULL if none.
  **/
 GtkWidget *
 gtk_menu_item_get_submenu (GtkMenuItem *menu_item)
diff --git a/gtk/gtkmenutoolbutton.c b/gtk/gtkmenutoolbutton.c
index c8cbf56..0c464e8 100644
--- a/gtk/gtkmenutoolbutton.c
+++ b/gtk/gtkmenutoolbutton.c
@@ -611,7 +611,8 @@ gtk_menu_tool_button_set_menu (GtkMenuToolButton *button,
  *
  * Gets the #GtkMenu associated with #GtkMenuToolButton.
  *
- * Return value: the #GtkMenu associated with #GtkMenuToolButton
+ * Return value: (transfer none): the #GtkMenu associated
+ *     with #GtkMenuToolButton
  *
  * Since: 2.6
  **/
diff --git a/gtk/gtkmountoperation.c b/gtk/gtkmountoperation.c
index 7837263..0747ed5 100644
--- a/gtk/gtkmountoperation.c
+++ b/gtk/gtkmountoperation.c
@@ -1450,7 +1450,7 @@ gtk_mount_operation_set_parent (GtkMountOperation *op,
  *
  * Gets the transient parent used by the #GtkMountOperation
  *
- * Returns: the transient parent for windows shown by @op
+ * Returns: (transfer none): the transient parent for windows shown by @op
  *
  * Since: 2.14
  */
@@ -1500,10 +1500,10 @@ gtk_mount_operation_set_screen (GtkMountOperation *op,
  * gtk_mount_operation_get_screen:
  * @op: a #GtkMountOperation
  *
- * Gets the screen on which windows of the #GtkMountOperation 
+ * Gets the screen on which windows of the #GtkMountOperation
  * will be shown.
  *
- * Returns: the screen on which windows of @op are shown
+ * Returns: (transfer none): the screen on which windows of @op are shown
  *
  * Since: 2.14
  */
diff --git a/gtk/gtknotebook.c b/gtk/gtknotebook.c
index 2780e2b..7d89c1c 100644
--- a/gtk/gtknotebook.c
+++ b/gtk/gtknotebook.c
@@ -7416,12 +7416,12 @@ gtk_notebook_get_tab_label_text (GtkNotebook *notebook,
  * gtk_notebook_get_menu_label:
  * @notebook: a #GtkNotebook
  * @child: a widget contained in a page of @notebook
- * 
+ *
  * Retrieves the menu label widget of the page containing @child.
- * 
- * Return value: the menu label, or %NULL if the
- *               notebook page does not have a menu label other
- *               than the default (the tab label).
+ *
+ * Return value: (transfer none): the menu label, or %NULL if the
+ *     notebook page does not have a menu label other than the
+ *     default (the tab label).
  **/
 GtkWidget*
 gtk_notebook_get_menu_label (GtkNotebook *notebook,
@@ -7433,14 +7433,14 @@ gtk_notebook_get_menu_label (GtkNotebook *notebook,
   g_return_val_if_fail (GTK_IS_WIDGET (child), NULL);
 
   list = CHECK_FIND_CHILD (notebook, child);
-  if (!list)  
+  if (!list)
     return NULL;
 
   if (GTK_NOTEBOOK_PAGE (list)->default_menu)
     return NULL;
 
   return GTK_NOTEBOOK_PAGE (list)->menu_label;
-}  
+}
 
 /**
  * gtk_notebook_set_menu_label:
@@ -7867,10 +7867,11 @@ gtk_notebook_get_group_id (GtkNotebook *notebook)
 /**
  * gtk_notebook_get_group:
  * @notebook: a #GtkNotebook
- * 
+ *
  * Gets the current group identificator pointer for @notebook.
- * 
- * Return Value: the group identificator, or %NULL if none is set.
+ *
+ * Return Value: (transfer none): the group identificator,
+ *     or %NULL if none is set.
  *
  * Since: 2.12
  *
@@ -8071,8 +8072,8 @@ gtk_notebook_set_tab_detachable (GtkNotebook *notebook,
  *
  * Gets one of the action widgets. See gtk_notebook_set_action_widget().
  *
- * Returns: The action widget with the given @pack_type or
- *     %NULL when this action widget has not been set
+ * Returns: (transfer none): The action widget with the given @pack_type
+ *     or %NULL when this action widget has not been set
  *
  * Since: 2.20
  */
diff --git a/gtk/gtkoffscreenwindow.c b/gtk/gtkoffscreenwindow.c
index 2b4207a..0fc1ddc 100644
--- a/gtk/gtkoffscreenwindow.c
+++ b/gtk/gtkoffscreenwindow.c
@@ -266,7 +266,8 @@ gtk_offscreen_window_new (void)
  * a #GdkPixmap.  If you need to keep this around over window
  * resizes then you should add a reference to it.
  *
- * Returns: A #GdkPixmap pointer to the offscreen pixmap, or %NULL.
+ * Returns: (transfer none): A #GdkPixmap pointer to the offscreen pixmap,
+ *     or %NULL.
  *
  * Since: 2.20
  */
@@ -287,7 +288,7 @@ gtk_offscreen_window_get_pixmap (GtkOffscreenWindow *offscreen)
  * and the application should unreference it once it is no longer
  * needed.
  *
- * Returns: A #GdkPixbuf pointer, or %NULL.
+ * Returns: (transfer full): A #GdkPixbuf pointer, or %NULL.
  *
  * Since: 2.20
  */
diff --git a/gtk/gtkpagesetup.c b/gtk/gtkpagesetup.c
index 0e8c153..6f0b448 100644
--- a/gtk/gtkpagesetup.c
+++ b/gtk/gtkpagesetup.c
@@ -99,10 +99,10 @@ gtk_page_setup_new (void)
 /**
  * gtk_page_setup_copy:
  * @other: the #GtkPageSetup to copy
- * 
+ *
  * Copies a #GtkPageSetup.
- * 
- * Return value: a copy of @other
+ *
+ * Return value: (transfer full): a copy of @other
  *
  * Since: 2.10
  */
diff --git a/gtk/gtkpaned.c b/gtk/gtkpaned.c
index a244686..f6225f0 100644
--- a/gtk/gtkpaned.c
+++ b/gtk/gtkpaned.c
@@ -1540,7 +1540,7 @@ gtk_paned_set_position (GtkPaned *paned,
  * 
  * Obtains the first child of the paned widget.
  * 
- * Return value: first child, or %NULL if it is not set.
+ * Return value: (transfer none): first child, or %NULL if it is not set.
  *
  * Since: 2.4
  **/
@@ -1558,7 +1558,7 @@ gtk_paned_get_child1 (GtkPaned *paned)
  * 
  * Obtains the second child of the paned widget.
  * 
- * Return value: second child, or %NULL if it is not set.
+ * Return value: (transfer none): second child, or %NULL if it is not set.
  *
  * Since: 2.4
  **/
@@ -2240,7 +2240,7 @@ gtk_paned_toggle_handle_focus (GtkPaned *paned)
  * enables the callback to distinguish between the window
  * of the paned, a child and the handle.
  *
- * Return value: the paned's handle window.
+ * Return value: (transfer none): the paned's handle window.
  *
  * Since: 2.20
  **/
diff --git a/gtk/gtkplug.c b/gtk/gtkplug.c
index 0a31d22..99af62a 100644
--- a/gtk/gtkplug.c
+++ b/gtk/gtkplug.c
@@ -307,7 +307,7 @@ gtk_plug_get_embedded (GtkPlug *plug)
  *
  * Retrieves the socket the plug is embedded in.
  *
- * Return value: the window of the socket, or %NULL
+ * Return value: (transfer none): the window of the socket, or %NULL
  *
  * Since: 2.14
  **/
@@ -540,7 +540,7 @@ gtk_plug_new (GdkNativeWindow socket_id)
 
 /**
  * gtk_plug_new_for_display:
- * @display : the #GdkDisplay on which @socket_id is displayed
+ * @display: the #GdkDisplay on which @socket_id is displayed
  * @socket_id: the XID of the socket's window.
  * 
  * Create a new plug widget inside the #GtkSocket identified by socket_id.
diff --git a/gtk/gtkprintcontext.c b/gtk/gtkprintcontext.c
index cdf04d7..d775bd7 100644
--- a/gtk/gtkprintcontext.c
+++ b/gtk/gtkprintcontext.c
@@ -273,7 +273,7 @@ gtk_print_context_get_cairo_context (GtkPrintContext *context)
  * Obtains the #GtkPageSetup that determines the page
  * dimensions of the #GtkPrintContext.
  *
- * Return value: the page setup of @context
+ * Return value: (transfer none): the page setup of @context
  *
  * Since: 2.10
  */
@@ -441,10 +441,10 @@ _gtk_print_context_set_hard_margins (GtkPrintContext *context,
  * gtk_print_context_get_pango_fontmap:
  * @context: a #GtkPrintContext
  *
- * Returns a #PangoFontMap that is suitable for use 
+ * Returns a #PangoFontMap that is suitable for use
  * with the #GtkPrintContext.
  *
- * Return value: the font map of @context
+ * Return value: (transfer none): the font map of @context
  *
  * Since: 2.10
  */
@@ -458,12 +458,12 @@ gtk_print_context_get_pango_fontmap (GtkPrintContext *context)
 
 /**
  * gtk_print_context_create_pango_context:
- * @context: a #GtkPrintContext 
+ * @context: a #GtkPrintContext
  *
  * Creates a new #PangoContext that can be used with the
  * #GtkPrintContext.
  *
- * Return value: a new Pango context for @context
+ * Return value: (transfer full): a new Pango context for @context
  * 
  * Since: 2.10
  */
@@ -497,7 +497,7 @@ gtk_print_context_create_pango_context (GtkPrintContext *context)
  * Creates a new #PangoLayout that is suitable for use
  * with the #GtkPrintContext.
  * 
- * Return value: a new Pango layout for @context
+ * Return value: (transfer full): a new Pango layout for @context
  *
  * Since: 2.10
  */
diff --git a/gtk/gtkprintoperation.c b/gtk/gtkprintoperation.c
index b6a77b4..7247b94 100644
--- a/gtk/gtkprintoperation.c
+++ b/gtk/gtkprintoperation.c
@@ -1401,10 +1401,10 @@ gtk_print_operation_set_default_page_setup (GtkPrintOperation *op,
  * gtk_print_operation_get_default_page_setup:
  * @op: a #GtkPrintOperation
  *
- * Returns the default page setup, see 
+ * Returns the default page setup, see
  * gtk_print_operation_set_default_page_setup().
  *
- * Returns: the default page setup 
+ * Returns: (transfer none): the default page setup
  *
  * Since: 2.10
  */
@@ -1457,15 +1457,15 @@ gtk_print_operation_set_print_settings (GtkPrintOperation *op,
 /**
  * gtk_print_operation_get_print_settings:
  * @op: a #GtkPrintOperation
- * 
- * Returns the current print settings. 
  *
- * Note that the return value is %NULL until either 
- * gtk_print_operation_set_print_settings() or 
+ * Returns the current print settings.
+ *
+ * Note that the return value is %NULL until either
+ * gtk_print_operation_set_print_settings() or
  * gtk_print_operation_run() have been called.
- * 
- * Return value: the current print settings of @op.
- * 
+ *
+ * Return value: (transfer none): the current print settings of @op.
+ *
  * Since: 2.10
  **/
 GtkPrintSettings *
diff --git a/gtk/gtkprintsettings.c b/gtk/gtkprintsettings.c
index 6d1c818..79d3bff 100644
--- a/gtk/gtkprintsettings.c
+++ b/gtk/gtkprintsettings.c
@@ -107,10 +107,10 @@ copy_hash_entry  (gpointer  key,
 /**
  * gtk_print_settings_copy:
  * @other: a #GtkPrintSettings
- * 
+ *
  * Copies a #GtkPrintSettings object.
- * 
- * Return value: a newly allocated copy of @other
+ *
+ * Return value: (transfer full): a newly allocated copy of @other
  *
  * Since: 2.10
  */
@@ -473,7 +473,7 @@ gtk_print_settings_set_int (GtkPrintSettings *settings,
 /**
  * gtk_print_settings_foreach:
  * @settings: a #GtkPrintSettings
- * @func: (scope call) the function to call
+ * @func: (scope call): the function to call
  * @user_data: user data for @func
  *
  * Calls @func for each key-value pair of @settings.
@@ -1402,16 +1402,16 @@ gtk_print_settings_set_print_pages (GtkPrintSettings *settings,
   
   gtk_print_settings_set (settings, GTK_PRINT_SETTINGS_PRINT_PAGES, str);
 }
-     
+
 /**
  * gtk_print_settings_get_page_ranges:
  * @settings: a #GtkPrintSettings
  * @num_ranges: return location for the length of the returned array
- * 
+ *
  * Gets the value of %GTK_PRINT_SETTINGS_PAGE_RANGES.
- * 
- * Return value: an array of #GtkPageRange<!-- -->s. Use g_free()
- *   to free the array when it is no longer needed. 
+ *
+ * Return value: (transfer full): an array of #GtkPageRange<!-- -->s.
+ *     Use g_free() to free the array when it is no longer needed.
  *
  * Since: 2.10
  */
diff --git a/gtk/gtkradiobutton.c b/gtk/gtkradiobutton.c
index df0066c..62a8eb3 100644
--- a/gtk/gtkradiobutton.c
+++ b/gtk/gtkradiobutton.c
@@ -306,12 +306,13 @@ gtk_radio_button_new_with_label (GSList      *group,
  * @group: the radio button group
  * @label: the text of the button, with an underscore in front of the
  *         mnemonic character
- * @returns: a new #GtkRadioButton
  *
  * Creates a new #GtkRadioButton containing a label, adding it to the same 
  * group as @group. The label will be created using 
  * gtk_label_new_with_mnemonic(), so underscores in @label indicate the 
  * mnemonic for the button.
+ *
+ * Returns: (transfer full): a new #GtkRadioButton
  **/
 GtkWidget*
 gtk_radio_button_new_with_mnemonic (GSList      *group,
@@ -374,11 +375,12 @@ gtk_radio_button_new_with_label_from_widget (GtkRadioButton *radio_group_member,
  * @radio_group_member: (allow-none): widget to get radio group from or %NULL
  * @label: the text of the button, with an underscore in front of the
  *         mnemonic character
- * @returns: a new #GtkRadioButton
  *
  * Creates a new #GtkRadioButton containing a label. The label
  * will be created using gtk_label_new_with_mnemonic(), so underscores
  * in @label indicate the mnemonic for the button.
+ *
+ * Returns: (transfer full): a new #GtkRadioButton
  **/
 GtkWidget*
 gtk_radio_button_new_with_mnemonic_from_widget (GtkRadioButton *radio_group_member,
diff --git a/gtk/gtkradiomenuitem.c b/gtk/gtkradiomenuitem.c
index 9bb5c11..eb02248 100644
--- a/gtk/gtkradiomenuitem.c
+++ b/gtk/gtkradiomenuitem.c
@@ -241,11 +241,11 @@ gtk_radio_menu_item_new_with_mnemonic (GSList *group,
 /**
  * gtk_radio_menu_item_new_from_widget:
  * @group: An existing #GtkRadioMenuItem
- * 
+ *
  * Creates a new #GtkRadioMenuItem adding it to the same group as @group.
- * 
- * Return value: The new #GtkRadioMenuItem
- * 
+ *
+ * Return value: (transfer full): The new #GtkRadioMenuItem
+ *
  * Since: 2.4
  **/
 GtkWidget *
@@ -273,8 +273,8 @@ gtk_radio_menu_item_new_from_widget (GtkRadioMenuItem *group)
  *
  * The new #GtkRadioMenuItem is added to the same group as @group.
  *
- * Return value: The new #GtkRadioMenuItem
- * 
+ * Return value: (transfer full): The new #GtkRadioMenuItem
+ *
  * Since: 2.4
  **/
 GtkWidget *
@@ -293,14 +293,14 @@ gtk_radio_menu_item_new_with_mnemonic_from_widget (GtkRadioMenuItem *group,
 
 /**
  * gtk_radio_menu_item_new_with_label_from_widget:
- * @group: an existing #GtkRadioMenuItem 
+ * @group: an existing #GtkRadioMenuItem
  * @label: the text for the label
  *
  * Creates a new GtkRadioMenuItem whose child is a simple GtkLabel.
  * The new #GtkRadioMenuItem is added to the same group as @group.
  *
- * Return value: The new #GtkRadioMenuItem
- * 
+ * Return value: (transfer full): The new #GtkRadioMenuItem
+ *
  * Since: 2.4
  **/
 GtkWidget *
@@ -317,6 +317,15 @@ gtk_radio_menu_item_new_with_label_from_widget (GtkRadioMenuItem *group,
   return gtk_radio_menu_item_new_with_label (list, label);
 }
 
+/**
+ * gtk_radio_menu_item_get_group:
+ * @radio_menu_item: a #GtkRadioMenuItem
+ *
+ * Returns the group to which the radio menu item belongs, as a #GList of
+ * #GtkRadioMenuItem. The list belongs to GTK+ and should not be freed.
+ *
+ * Returns: (transfer none): the group of @radio_menu_item
+ */
 GSList*
 gtk_radio_menu_item_get_group (GtkRadioMenuItem *radio_menu_item)
 {
diff --git a/gtk/gtkradiotoolbutton.c b/gtk/gtkradiotoolbutton.c
index 1aa37da..6de690d 100644
--- a/gtk/gtkradiotoolbutton.c
+++ b/gtk/gtkradiotoolbutton.c
@@ -164,11 +164,11 @@ gtk_radio_tool_button_new_from_stock (GSList      *group,
 /**
  * gtk_radio_tool_button_new_from_widget:
  * @group: An existing #GtkRadioToolButton
- * 
+ *
  * Creates a new #GtkRadioToolButton adding it to the same group as @gruup
- * 
- * Return value: The new #GtkRadioToolButton
- * 
+ *
+ * Return value: (transfer full): The new #GtkRadioToolButton
+ *
  * Since: 2.4
  **/
 GtkToolItem *
@@ -187,14 +187,14 @@ gtk_radio_tool_button_new_from_widget (GtkRadioToolButton *group)
 /**
  * gtk_radio_tool_button_new_with_stock_from_widget:
  * @group: An existing #GtkRadioToolButton.
- * @stock_id: the name of a stock item 
- * 
+ * @stock_id: the name of a stock item
+ *
  * Creates a new #GtkRadioToolButton adding it to the same group as @group.
  * The new #GtkRadioToolButton will contain an icon and label from the
  * stock item indicated by @stock_id.
- * 
- * Return value: A new #GtkRadioToolButton
- * 
+ *
+ * Return value: (transfer full): A new #GtkRadioToolButton
+ *
  * Since: 2.4
  **/
 GtkToolItem *
@@ -222,11 +222,11 @@ get_radio_button (GtkRadioToolButton *button)
  * @button: a #GtkRadioToolButton
  *
  * Returns the radio button group @button belongs to.
- * 
- * Return value: The group @button belongs to.
- * 
+ *
+ * Return value: (transfer none): The group @button belongs to.
+ *
  * Since: 2.4
- **/
+ */
 GSList *
 gtk_radio_tool_button_get_group (GtkRadioToolButton *button)
 {
diff --git a/gtk/gtkrange.c b/gtk/gtkrange.c
index 8669b41..664d387 100644
--- a/gtk/gtkrange.c
+++ b/gtk/gtkrange.c
@@ -733,7 +733,7 @@ gtk_range_init (GtkRange *range)
  * The return value does not have a reference added, so should not
  * be unreferenced.
  * 
- * Return value: a #GtkAdjustment
+ * Return value: (transfer none): a #GtkAdjustment
  **/
 GtkAdjustment*
 gtk_range_get_adjustment (GtkRange *range)
@@ -1417,7 +1417,7 @@ gtk_range_set_fill_level (GtkRange *range,
 
 /**
  * gtk_range_get_fill_level:
- * @range : A #GtkRange
+ * @range: A #GtkRange
  *
  * Gets the current position of the fill level indicator.
  *
diff --git a/gtk/gtkrc.c b/gtk/gtkrc.c
index 26011f6..e60437c 100644
--- a/gtk/gtkrc.c
+++ b/gtk/gtkrc.c
@@ -601,13 +601,13 @@ gtk_rc_set_default_files (gchar **filenames)
 
 /**
  * gtk_rc_get_default_files:
- * 
+ *
  * Retrieves the current list of RC files that will be parsed
  * at the end of gtk_init().
- * 
- * Return value: A %NULL-terminated array of filenames. This memory
- * is owned by GTK+ and must not be freed by the application.
- * If you want to store this information, you should make a copy.
+ *
+ * Return value: (transfer none): A %NULL-terminated array of filenames.
+ *     This memory is owned by GTK+ and must not be freed by the application.
+ *     If you want to store this information, you should make a copy.
  **/
 gchar **
 gtk_rc_get_default_files (void)
@@ -1250,12 +1250,12 @@ gtk_rc_style_new (void)
 /**
  * gtk_rc_style_copy:
  * @orig: the style to copy
- * 
+ *
  * Makes a copy of the specified #GtkRcStyle. This function
  * will correctly copy an RC style that is a member of a class
  * derived from #GtkRcStyle.
- * 
- * Return value: the resulting #GtkRcStyle
+ *
+ * Return value: (transfer full): the resulting #GtkRcStyle
  **/
 GtkRcStyle *
 gtk_rc_style_copy (GtkRcStyle *orig)
@@ -2028,12 +2028,12 @@ gtk_rc_get_style (GtkWidget *widget)
 /**
  * gtk_rc_get_style_by_paths:
  * @settings: a #GtkSettings object
- * @widget_path: (allow-none): the widget path to use when looking up the style, or %NULL
- *               if no matching against the widget path should be done
- * @class_path: (allow-none): the class path to use when looking up the style, or %NULL
- *               if no matching against the class path should be done.
+ * @widget_path: (allow-none): the widget path to use when looking up the
+ *     style, or %NULL if no matching against the widget path should be done
+ * @class_path: (allow-none): the class path to use when looking up the style,
+ *     or %NULL if no matching against the class path should be done.
  * @type: a type that will be used along with parent types of this type
- *        when matching against class styles, or #G_TYPE_NONE
+ *     when matching against class styles, or #G_TYPE_NONE
  *
  * Creates up a #GtkStyle from styles defined in a RC file by providing
  * the raw components used in matching. This function may be useful
@@ -2050,11 +2050,11 @@ gtk_rc_get_style (GtkWidget *widget)
  *                             G_OBJECT_TYPE (widget));
  * ]|
  * 
- * Return value: A style created by matching with the supplied paths,
- *   or %NULL if nothing matching was specified and the default style should
- *   be used. The returned value is owned by GTK+ as part of an internal cache,
- *   so you must call g_object_ref() on the returned value if you want to
- *   keep a reference to it.
+ * Return value: (transfer none): A style created by matching with the
+ *     supplied paths, or %NULL if nothing matching was specified and the
+ *     default style should be used. The returned value is owned by GTK+
+ *     as part of an internal cache, so you must call g_object_ref() on
+ *     the returned value if you want to keep a reference to it.
  **/
 GtkStyle *
 gtk_rc_get_style_by_paths (GtkSettings *settings,
diff --git a/gtk/gtkrecentchooser.c b/gtk/gtkrecentchooser.c
index bb84ef5..de60dc2 100644
--- a/gtk/gtkrecentchooser.c
+++ b/gtk/gtkrecentchooser.c
@@ -925,8 +925,8 @@ gtk_recent_chooser_get_items (GtkRecentChooser *chooser)
  *
  * Since the returned array is %NULL terminated, @length may be %NULL.
  * 
- * Return value: A newly allocated, %NULL terminated array of strings. Use
- *   g_strfreev() to free it.
+ * Return value: (transfer full): A newly allocated, %NULL-terminated array
+ *     of strings. Use g_strfreev() to free it.
  *
  * Since: 2.10
  */
@@ -1056,7 +1056,7 @@ gtk_recent_chooser_set_filter (GtkRecentChooser *chooser,
  * Gets the #GtkRecentFilter object currently used by @chooser to affect
  * the display of the recently used resources.
  *
- * Return value: a #GtkRecentFilter object.
+ * Return value: (transfer none): a #GtkRecentFilter object.
  *
  * Since: 2.10
  */
diff --git a/gtk/gtkrecentmanager.c b/gtk/gtkrecentmanager.c
index b343920..7bcf839 100644
--- a/gtk/gtkrecentmanager.c
+++ b/gtk/gtkrecentmanager.c
@@ -1922,8 +1922,9 @@ gtk_recent_info_get_application_info (GtkRecentInfo  *info,
  *
  * Retrieves the list of applications that have registered this resource.
  *
- * Return value: (array length=length zero-terminated=1): a newly allocated
- *  %NULL-terminated array of strings. Use g_strfreev() to free it.
+ * Return value: (array length=length zero-terminated=1) (transfer full):
+ *     a newly allocated %NULL-terminated array of strings.
+ *     Use g_strfreev() to free it.
  *
  * Since: 2.10
  */
@@ -2082,8 +2083,8 @@ get_icon_fallback (const gchar *icon_name,
  *
  * Retrieves the icon of size @size associated to the resource MIME type.
  *
- * Return value: a #GdkPixbuf containing the icon, or %NULL. Use
- *   g_object_unref() when finished using the icon.
+ * Return value: (transfer full): a #GdkPixbuf containing the icon,
+ *     or %NULL. Use g_object_unref() when finished using the icon.
  *
  * Since: 2.10
  */
@@ -2419,8 +2420,9 @@ gtk_recent_info_get_age (GtkRecentInfo *info)
  * array of returned group names will be %NULL terminated, so length might
  * optionally be %NULL.
  *
- * Return value:  (array length=length zero-terminated=1): a newly allocated
- * %NULL terminated array of strings.  Use g_strfreev() to free it.
+ * Return value:  (array length=length zero-terminated=1) (transfer full):
+ *     a newly allocated %NULL terminated array of strings.
+ *     Use g_strfreev() to free it.
  *
  * Since: 2.10
  */
diff --git a/gtk/gtkscale.c b/gtk/gtkscale.c
index 9e92d9c..d04ee2a 100644
--- a/gtk/gtkscale.c
+++ b/gtk/gtkscale.c
@@ -1263,13 +1263,13 @@ gtk_scale_finalize (GObject *object)
  * gtk_scale_get_layout:
  * @scale: A #GtkScale
  *
- * Gets the #PangoLayout used to display the scale. 
- * The returned object is owned by the scale so does 
- * not need to be freed by the caller. 
+ * Gets the #PangoLayout used to display the scale. The returned
+ * object is owned by the scale so does not need to be freed by
+ * the caller.
+ *
+ * Return value: (transfer none): the #PangoLayout for this scale,
+ *     or %NULL if the #GtkScale:draw-value property is %FALSE.
  *
- * Return value: the #PangoLayout for this scale, or %NULL 
- *    if the #GtkScale:draw-value property is %FALSE.
- *   
  * Since: 2.4
  */
 PangoLayout *
diff --git a/gtk/gtkscalebutton.c b/gtk/gtkscalebutton.c
index 7259622..fbfca35 100644
--- a/gtk/gtkscalebutton.c
+++ b/gtk/gtkscalebutton.c
@@ -660,7 +660,7 @@ gtk_scale_button_set_icons (GtkScaleButton  *button,
  * Gets the #GtkAdjustment associated with the #GtkScaleButton's scale.
  * See gtk_range_get_adjustment() for details.
  *
- * Returns: the adjustment associated with the scale
+ * Returns: (transfer none): the adjustment associated with the scale
  *
  * Since: 2.12
  */
@@ -752,7 +752,7 @@ gtk_scale_button_set_orientation (GtkScaleButton *button,
  *
  * Retrieves the plus button of the #GtkScaleButton.
  *
- * Returns: the plus button of the #GtkScaleButton.
+ * Returns: (transfer none): the plus button of the #GtkScaleButton
  *
  * Since: 2.14
  */
@@ -770,7 +770,7 @@ gtk_scale_button_get_plus_button (GtkScaleButton *button)
  *
  * Retrieves the minus button of the #GtkScaleButton.
  *
- * Returns: the minus button of the #GtkScaleButton.
+ * Returns: (transfer none): the minus button of the #GtkScaleButton
  *
  * Since: 2.14
  */
@@ -788,7 +788,7 @@ gtk_scale_button_get_minus_button (GtkScaleButton *button)
  *
  * Retrieves the popup of the #GtkScaleButton.
  *
- * Returns: the popup of the #GtkScaleButton
+ * Returns: (transfer none): the popup of the #GtkScaleButton
  *
  * Since: 2.14
  */
diff --git a/gtk/gtkscrolledwindow.c b/gtk/gtkscrolledwindow.c
index 5a3cd91..ffb151d 100644
--- a/gtk/gtkscrolledwindow.c
+++ b/gtk/gtkscrolledwindow.c
@@ -540,7 +540,7 @@ gtk_scrolled_window_set_vadjustment (GtkScrolledWindow *scrolled_window,
  * horizontal scrollbar to the child widget's horizontal scroll
  * functionality.
  *
- * Returns: the horizontal #GtkAdjustment
+ * Returns: (transfer none): the horizontal #GtkAdjustment
  */
 GtkAdjustment*
 gtk_scrolled_window_get_hadjustment (GtkScrolledWindow *scrolled_window)
@@ -557,10 +557,9 @@ gtk_scrolled_window_get_hadjustment (GtkScrolledWindow *scrolled_window)
  * @scrolled_window: a #GtkScrolledWindow
  * 
  * Returns the vertical scrollbar's adjustment, used to connect the
- * vertical scrollbar to the child widget's vertical scroll
- * functionality.
+ * vertical scrollbar to the child widget's vertical scroll functionality.
  * 
- * Returns: the vertical #GtkAdjustment
+ * Returns: (transfer none): the vertical #GtkAdjustment
  */
 GtkAdjustment*
 gtk_scrolled_window_get_vadjustment (GtkScrolledWindow *scrolled_window)
@@ -575,11 +574,11 @@ gtk_scrolled_window_get_vadjustment (GtkScrolledWindow *scrolled_window)
 /**
  * gtk_scrolled_window_get_hscrollbar:
  * @scrolled_window: a #GtkScrolledWindow
- * 
+ *
  * Returns the horizontal scrollbar of @scrolled_window.
  *
- * Returns: the horizontal scrollbar of the scrolled window, or 
- *  %NULL if it does not have one.
+ * Returns: (transfer none): the horizontal scrollbar of the scrolled window,
+ *     or %NULL if it does not have one.
  *
  * Since: 2.8
  */
@@ -597,8 +596,8 @@ gtk_scrolled_window_get_hscrollbar (GtkScrolledWindow *scrolled_window)
  * 
  * Returns the vertical scrollbar of @scrolled_window.
  *
- * Returns: the vertical scrollbar of the scrolled window, or
- *  %NULL if it does not have one.
+ * Returns: (transfer none): the vertical scrollbar of the scrolled window,
+ *     or %NULL if it does not have one.
  *
  * Since: 2.8
  */
diff --git a/gtk/gtkselection.c b/gtk/gtkselection.c
index fa20852..ba0813e 100644
--- a/gtk/gtkselection.c
+++ b/gtk/gtkselection.c
@@ -1226,7 +1226,7 @@ gtk_selection_data_get_length (GtkSelectionData *selection_data)
  *
  * Retrieves the display of the selection.
  *
- * Returns: the display of the selection.
+ * Returns: (transfer none): the display of the selection.
  *
  * Since: 2.14
  **/
@@ -1660,7 +1660,7 @@ gtk_selection_data_set_pixbuf (GtkSelectionData *selection_data,
  * 
  * Gets the contents of the selection data as a #GdkPixbuf.
  * 
- * Return value: if the selection data contained a recognized
+ * Return value: (transfer full): if the selection data contained a recognized
  *   image type and it could be converted to a #GdkPixbuf, a 
  *   newly allocated pixbuf is returned, otherwise %NULL.
  *   If the result is non-%NULL it must be freed with g_object_unref().
diff --git a/gtk/gtksettings.c b/gtk/gtksettings.c
index 774de19..de96b91 100644
--- a/gtk/gtksettings.c
+++ b/gtk/gtksettings.c
@@ -1040,11 +1040,11 @@ gtk_settings_finalize (GObject *object)
 
 /**
  * gtk_settings_get_for_screen:
- * @screen : a #GdkScreen.
- * 
+ * @screen: a #GdkScreen.
+ *
  * Gets the #GtkSettings object for @screen, creating it if necessary.
  *
- * Return value: a #GtkSettings object.
+ * Return value: (transfer none): a #GtkSettings object.
  *
  * Since: 2.2
  */
@@ -1052,9 +1052,9 @@ GtkSettings*
 gtk_settings_get_for_screen (GdkScreen *screen)
 {
   GtkSettings *settings;
-  
+
   g_return_val_if_fail (GDK_IS_SCREEN (screen), NULL);
-  
+
   settings = g_object_get_data (G_OBJECT (screen), "gtk-settings");
   if (!settings)
     {
diff --git a/gtk/gtksocket.c b/gtk/gtksocket.c
index 3531af4..e16755a 100644
--- a/gtk/gtksocket.c
+++ b/gtk/gtksocket.c
@@ -378,7 +378,7 @@ gtk_socket_get_id (GtkSocket *socket)
  * Retrieves the window of the plug. Use this to check if the plug has
  * been created inside of the socket.
  *
- * Return value: the window of the plug if available, or %NULL
+ * Return value: (transfer none): the window of the plug if available, or %NULL
  *
  * Since:  2.14
  **/
diff --git a/gtk/gtkspinbutton.c b/gtk/gtkspinbutton.c
index 792f554..53734cc 100644
--- a/gtk/gtkspinbutton.c
+++ b/gtk/gtkspinbutton.c
@@ -1786,7 +1786,7 @@ gtk_spin_button_set_adjustment (GtkSpinButton *spin_button,
  * 
  * Get the adjustment associated with a #GtkSpinButton
  * 
- * Return value: the #GtkAdjustment of @spin_button
+ * Return value: (transfer none): the #GtkAdjustment of @spin_button
  **/
 GtkAdjustment *
 gtk_spin_button_get_adjustment (GtkSpinButton *spin_button)
diff --git a/gtk/gtkstatusbar.c b/gtk/gtkstatusbar.c
index 81d28d0..3edecbe 100644
--- a/gtk/gtkstatusbar.c
+++ b/gtk/gtkstatusbar.c
@@ -588,7 +588,7 @@ gtk_statusbar_get_has_resize_grip (GtkStatusbar *statusbar)
  *
  * Retrieves the box containing the label widget.
  *
- * Returns: a #GtkBox
+ * Returns: (transfer none): a #GtkBox
  *
  * Since: 2.20
  */
diff --git a/gtk/gtkstatusicon.c b/gtk/gtkstatusicon.c
index 77b4365..65edd6e 100644
--- a/gtk/gtkstatusicon.c
+++ b/gtk/gtkstatusicon.c
@@ -2037,7 +2037,8 @@ gtk_status_icon_get_storage_type (GtkStatusIcon *status_icon)
  * The caller of this function does not own a reference to the
  * returned pixbuf.
  * 
- * Return value: the displayed pixbuf, or %NULL if the image is empty.
+ * Return value: (transfer none): the displayed pixbuf,
+ *     or %NULL if the image is empty.
  *
  * Since: 2.10
  **/
@@ -2136,7 +2137,7 @@ gtk_status_icon_get_icon_name (GtkStatusIcon *status_icon)
  *
  * If this function fails, @icon is left unchanged;
  *
- * Returns: the displayed icon, or %NULL if the image is empty
+ * Returns: (transfer none): the displayed icon, or %NULL if the image is empty
  *
  * Since: 2.14
  **/
@@ -2205,13 +2206,13 @@ gtk_status_icon_set_screen (GtkStatusIcon *status_icon,
 #endif
 }
 
-/** 
+/**
  * gtk_status_icon_get_screen:
  * @status_icon: a #GtkStatusIcon
  *
  * Returns the #GdkScreen associated with @status_icon.
  *
- * Return value: a #GdkScreen.
+ * Return value: (transfer none): a #GdkScreen.
  *
  * Since: 2.12
  */
@@ -2220,7 +2221,7 @@ gtk_status_icon_get_screen (GtkStatusIcon *status_icon)
 {
   g_return_val_if_fail (GTK_IS_STATUS_ICON (status_icon), NULL);
 
-#ifdef GDK_WINDOWING_X11   
+#ifdef GDK_WINDOWING_X11
   return gtk_window_get_screen (GTK_WINDOW (status_icon->priv->tray_icon));
 #else
   return gdk_screen_get_default ();
diff --git a/gtk/gtkstyle.c b/gtk/gtkstyle.c
index fce7ddf..8654d1f 100644
--- a/gtk/gtkstyle.c
+++ b/gtk/gtkstyle.c
@@ -645,7 +645,7 @@ gtk_style_finalize (GObject *object)
  *
  * Creates a copy of the passed in #GtkStyle object.
  *
- * Returns: a copy of @style
+ * Returns: (transfer full): a copy of @style
  */
 GtkStyle*
 gtk_style_copy (GtkStyle *style)
@@ -703,12 +703,12 @@ gtk_style_new (void)
  *
  * Attaches a style to a window; this process allocates the
  * colors and creates the GC's for the style - it specializes
- * it to a particular visual and colormap. The process may 
- * involve the creation of a new style if the style has already 
+ * it to a particular visual and colormap. The process may
+ * involve the creation of a new style if the style has already
  * been attached to a window with a different style and colormap.
  *
- * Since this function may return a new object, you have to use it 
- * in the following way: 
+ * Since this function may return a new object, you have to use it
+ * in the following way:
  * <literal>style = gtk_style_attach (style, window)</literal>
  *
  * Returns: Either @style, or a newly-created #GtkStyle.
@@ -2173,11 +2173,13 @@ gtk_style_real_set_background (GtkStyle    *style,
  *     don't scale.
  * @widget: (allow-none): the widget
  * @detail: (allow-none): a style detail
- * @returns: a newly-created #GdkPixbuf containing the rendered icon
  *
  * Renders the icon specified by @source at the given @size
  * according to the given parameters and returns the result in a
  * pixbuf.
+ *
+ * Return value: (transfer full): a newly-created #GdkPixbuf
+ *     containing the rendered icon
  */
 GdkPixbuf *
 gtk_style_render_icon (GtkStyle            *style,
diff --git a/gtk/gtktextbuffer.c b/gtk/gtktextbuffer.c
index cfe5c16..99a574f 100644
--- a/gtk/gtktextbuffer.c
+++ b/gtk/gtktextbuffer.c
@@ -1962,7 +1962,7 @@ gtk_text_buffer_insert_child_anchor (GtkTextBuffer      *buffer,
  * owned by the buffer; no reference count is returned to
  * the caller of gtk_text_buffer_create_child_anchor().
  * 
- * Return value: the created child anchor
+ * Return value: (transfer none): the created child anchor
  **/
 GtkTextChildAnchor*
 gtk_text_buffer_create_child_anchor (GtkTextBuffer *buffer,
@@ -2433,7 +2433,6 @@ gtk_text_buffer_select_range (GtkTextBuffer     *buffer,
  * @first_property_name: (allow-none): name of first property to set, or %NULL
  * @Varargs: %NULL-terminated list of property names and values
  *
- *
  * Creates a tag and adds it to the tag table for @buffer.
  * Equivalent to calling gtk_text_tag_new() and then adding the
  * tag to the buffer's tag table. The returned tag is owned by
@@ -2447,8 +2446,8 @@ gtk_text_buffer_select_range (GtkTextBuffer     *buffer,
  * The @first_property_name argument and subsequent arguments are a list
  * of properties to set on the tag, as with g_object_set().
  *
- * Return value: a new tag
- **/
+ * Return value: (transfer none): a new tag
+ */
 GtkTextTag*
 gtk_text_buffer_create_tag (GtkTextBuffer *buffer,
                             const gchar   *tag_name,
@@ -3736,10 +3735,10 @@ remove_all_selection_clipboards (GtkTextBuffer *buffer)
  *                     at the cursor
  * @default_editable: whether the buffer is editable by default
  *
- * Pastes the contents of a clipboard at the insertion point, or at
- * @override_location. (Note: pasting is asynchronous, that is, we'll 
- * ask for the paste data and return, and at some point later after 
- * the main loop runs, the paste data will be inserted.)
+ * Pastes the contents of a clipboard at the insertion point, or
+ * at @override_location. (Note: pasting is asynchronous, that is,
+ * we'll ask for the paste data and return, and at some point later
+ * after the main loop runs, the paste data will be inserted.)
  **/
 void
 gtk_text_buffer_paste_clipboard (GtkTextBuffer *buffer,
diff --git a/gtk/gtktextiter.c b/gtk/gtktextiter.c
index 5167cda..85beaf1 100644
--- a/gtk/gtktextiter.c
+++ b/gtk/gtktextiter.c
@@ -1002,7 +1002,7 @@ gtk_text_iter_get_pixbuf (const GtkTextIter *iter)
  * anchor is returned (with no new reference count added). Otherwise,
  * %NULL is returned.
  *
- * Return value: the anchor at @iter
+ * Return value: (transfer none): the anchor at @iter
  **/
 GtkTextChildAnchor*
 gtk_text_iter_get_child_anchor (const GtkTextIter *iter)
@@ -3239,7 +3239,7 @@ gtk_text_iter_forward_word_ends (GtkTextIter      *iter,
 }
 
 /**
- * gtk_text_iter_backward_word_starts
+ * gtk_text_iter_backward_word_starts:
  * @iter: a #GtkTextIter
  * @count: number of times to move
  * 
@@ -3317,7 +3317,7 @@ gtk_text_iter_forward_visible_word_ends (GtkTextIter *iter,
 }
 
 /**
- * gtk_text_iter_backward_visible_word_starts
+ * gtk_text_iter_backward_visible_word_starts:
  * @iter: a #GtkTextIter
  * @count: number of times to move
  * 
@@ -4306,7 +4306,7 @@ matches_pred (GtkTextIter *iter,
 /**
  * gtk_text_iter_forward_find_char:
  * @iter: a #GtkTextIter
- * @pred: a function to be called on each character
+ * @pred: (scope call): a function to be called on each character
  * @user_data: user data for @pred
  * @limit: (allow-none): search limit, or %NULL for none 
  * 
@@ -4344,7 +4344,7 @@ gtk_text_iter_forward_find_char (GtkTextIter         *iter,
 /**
  * gtk_text_iter_backward_find_char:
  * @iter: a #GtkTextIter
- * @pred: function to be called on each character
+ * @pred: (scope call): function to be called on each character
  * @user_data: user data for @pred
  * @limit: (allow-none): search limit, or %NULL for none
  * 
diff --git a/gtk/gtktexttagtable.c b/gtk/gtktexttagtable.c
index 25f1809..25fbf27 100644
--- a/gtk/gtktexttagtable.c
+++ b/gtk/gtktexttagtable.c
@@ -399,7 +399,7 @@ list_foreach (gpointer data, gpointer user_data)
 /**
  * gtk_text_tag_table_foreach:
  * @table: a #GtkTextTagTable
- * @func: a function to call on each tag
+ * @func: (scope call): a function to call on each tag
  * @data: user data
  *
  * Calls @func on each tag in @table, with user data @data.
diff --git a/gtk/gtktoolbar.c b/gtk/gtktoolbar.c
index b56abdf..17e2f8f 100644
--- a/gtk/gtktoolbar.c
+++ b/gtk/gtktoolbar.c
@@ -2997,10 +2997,10 @@ gtk_toolbar_get_n_items (GtkToolbar *toolbar)
  *
  * Returns the @n<!-- -->'th item on @toolbar, or %NULL if the
  * toolbar does not contain an @n<!-- -->'th item.
- * 
- * Return value: The @n<!-- -->'th #GtkToolItem on @toolbar, or %NULL if there
- * isn't an @n<!-- -->'th item.
- * 
+ *
+ * Return value: (transfer none): The @n<!-- -->'th #GtkToolItem on @toolbar,
+ *     or %NULL if there isn't an @n<!-- -->'th item.
+ *
  * Since: 2.4
  **/
 GtkToolItem *
diff --git a/gtk/gtktoolbutton.c b/gtk/gtktoolbutton.c
index 207c464..a67ff13 100644
--- a/gtk/gtktoolbutton.c
+++ b/gtk/gtktoolbutton.c
@@ -1252,12 +1252,13 @@ gtk_tool_button_set_label_widget (GtkToolButton *button,
 /**
  * gtk_tool_button_get_label_widget:
  * @button: a #GtkToolButton
- * 
- * Returns the widget used as label on @button. See
- * gtk_tool_button_set_label_widget().
- * 
- * Return value: The widget used as label on @button, or %NULL.
- * 
+ *
+ * Returns the widget used as label on @button.
+ * See gtk_tool_button_set_label_widget().
+ *
+ * Return value: (transfer none): The widget used as label
+ *     on @button, or %NULL.
+ *
  * Since: 2.4
  **/
 GtkWidget *
@@ -1271,12 +1272,13 @@ gtk_tool_button_get_label_widget (GtkToolButton *button)
 /**
  * gtk_tool_button_get_icon_widget:
  * @button: a #GtkToolButton
- * 
- * Return the widget used as icon widget on @button. See
- * gtk_tool_button_set_icon_widget().
- * 
- * Return value: The widget used as icon on @button, or %NULL.
- * 
+ *
+ * Return the widget used as icon widget on @button.
+ * See gtk_tool_button_set_icon_widget().
+ *
+ * Return value: (transfer none): The widget used as icon
+ *     on @button, or %NULL.
+ *
  * Since: 2.4
  **/
 GtkWidget *
diff --git a/gtk/gtktoolitem.c b/gtk/gtktoolitem.c
index 606ce1e..244f8c1 100644
--- a/gtk/gtktoolitem.c
+++ b/gtk/gtktoolitem.c
@@ -913,10 +913,11 @@ gtk_tool_item_get_text_orientation (GtkToolItem *tool_item)
  * gtk_tool_item_get_text_size_group:
  * @tool_item: a #GtkToolItem
  *
- * Returns the size group used for labels in @tool_item. Custom subclasses of
- * #GtkToolItem should call this function and use the size group for labels.
+ * Returns the size group used for labels in @tool_item.
+ * Custom subclasses of #GtkToolItem should call this function
+ * and use the size group for labels.
  *
- * Return value: a #GtkSizeGroup
+ * Return value: (transfer none): a #GtkSizeGroup
  *
  * Since: 2.20
  */
@@ -1355,20 +1356,21 @@ gtk_tool_item_retrieve_proxy_menu_item (GtkToolItem *tool_item)
 
 /**
  * gtk_tool_item_get_proxy_menu_item:
- * @tool_item: a #GtkToolItem 
+ * @tool_item: a #GtkToolItem
  * @menu_item_id: a string used to identify the menu item
- * 
+ *
  * If @menu_item_id matches the string passed to
  * gtk_tool_item_set_proxy_menu_item() return the corresponding #GtkMenuItem.
  *
- * Custom subclasses of #GtkToolItem should use this function to update
- * their menu item when the #GtkToolItem changes. That the
- * @menu_item_id<!-- -->s must match ensures that a #GtkToolItem will not
- * inadvertently change a menu item that they did not create.
- * 
- * Return value: The #GtkMenuItem passed to
- * gtk_tool_item_set_proxy_menu_item(), if the @menu_item_id<!-- -->s match.
- * 
+ * Custom subclasses of #GtkToolItem should use this function to
+ * update their menu item when the #GtkToolItem changes. That the
+ * @menu_item_id<!-- -->s must match ensures that a #GtkToolItem
+ * will not inadvertently change a menu item that they did not create.
+ *
+ * Return value: (transfer none): The #GtkMenuItem passed to
+ *     gtk_tool_item_set_proxy_menu_item(), if the @menu_item_id<!-- -->s
+ *     match.
+ *
  * Since: 2.4
  **/
 GtkWidget *
diff --git a/gtk/gtktoolitemgroup.c b/gtk/gtktoolitemgroup.c
index e831468..8574f26 100644
--- a/gtk/gtktoolitemgroup.c
+++ b/gtk/gtktoolitemgroup.c
@@ -1952,7 +1952,7 @@ gtk_tool_item_group_get_label (GtkToolItemGroup *group)
  * Gets the label widget of @group.
  * See gtk_tool_item_group_set_label_widget().
  *
- * Returns: the label widget of @group
+ * Returns: (transfer none): the label widget of @group
  *
  * Since: 2.20
  */
@@ -2157,7 +2157,7 @@ gtk_tool_item_group_get_n_items (GtkToolItemGroup *group)
  *
  * Gets the tool item at @index in group.
  *
- * Returns: the #GtkToolItem at index
+ * Returns: (transfer none): the #GtkToolItem at index
  *
  * Since: 2.20
  */
@@ -2182,7 +2182,7 @@ gtk_tool_item_group_get_nth_item (GtkToolItemGroup *group,
  *
  * Gets the tool item at position (x, y).
  *
- * Returns: the #GtkToolItem at position (x, y)
+ * Returns: (transfer none): the #GtkToolItem at position (x, y)
  *
  * Since: 2.20
  */
diff --git a/gtk/gtktreemodel.c b/gtk/gtktreemodel.c
index 32a4e6f..d5b2f12 100644
--- a/gtk/gtktreemodel.c
+++ b/gtk/gtktreemodel.c
@@ -1628,11 +1628,11 @@ gtk_tree_model_foreach_helper (GtkTreeModel            *model,
 /**
  * gtk_tree_model_foreach:
  * @model: A #GtkTreeModel
- * @func: A function to be called on each row
+ * @func: (scope call): A function to be called on each row
  * @user_data: User data to passed to func.
- * 
- * Calls func on each node in model in a depth-first fashion.  
- * If @func returns %TRUE, then the tree ceases to be walked, and 
+ *
+ * Calls func on each node in model in a depth-first fashion.
+ * If @func returns %TRUE, then the tree ceases to be walked, and
  * gtk_tree_model_foreach() returns.
  **/
 void
@@ -2047,7 +2047,7 @@ gtk_tree_row_reference_get_path (GtkTreeRowReference *reference)
  *
  * Returns the model that the row reference is monitoring.
  *
- * Return value: the model
+ * Return value: (transfer none): the model
  *
  * Since: 2.8
  */
diff --git a/gtk/gtktreemodelsort.c b/gtk/gtktreemodelsort.c
index 4fe4f34..4d085f4 100644
--- a/gtk/gtktreemodelsort.c
+++ b/gtk/gtktreemodelsort.c
@@ -324,10 +324,10 @@ gtk_tree_model_sort_drag_source_init (GtkTreeDragSourceIface *iface)
  *
  * Creates a new #GtkTreeModel, with @child_model as the child model.
  *
- * Return value: A new #GtkTreeModel.
+ * Return value: (transfer full): A new #GtkTreeModel.
  */
 GtkTreeModel *
-gtk_tree_model_sort_new_with_model (GtkTreeModel      *child_model)
+gtk_tree_model_sort_new_with_model (GtkTreeModel *child_model)
 {
   GtkTreeModel *retval;
 
@@ -1989,7 +1989,7 @@ gtk_tree_model_sort_set_model (GtkTreeModelSort *tree_model_sort,
  *
  * Returns the model the #GtkTreeModelSort is sorting.
  *
- * Return value: the "child model" being sorted
+ * Return value: (transfer none): the "child model" being sorted
  **/
 GtkTreeModel *
 gtk_tree_model_sort_get_model (GtkTreeModelSort *tree_model)
diff --git a/gtk/gtktreeview.c b/gtk/gtktreeview.c
index cbb2842..44eb25d 100644
--- a/gtk/gtktreeview.c
+++ b/gtk/gtktreeview.c
@@ -10888,8 +10888,8 @@ gtk_tree_view_get_selection (GtkTreeView *tree_view)
  *
  * Gets the #GtkAdjustment currently being used for the horizontal aspect.
  *
- * Return value: A #GtkAdjustment object, or %NULL if none is currently being
- * used.
+ * Return value: (transfer none): A #GtkAdjustment object, or %NULL
+ *     if none is currently being used.
  **/
 GtkAdjustment *
 gtk_tree_view_get_hadjustment (GtkTreeView *tree_view)
@@ -10928,8 +10928,8 @@ gtk_tree_view_set_hadjustment (GtkTreeView   *tree_view,
  *
  * Gets the #GtkAdjustment currently being used for the vertical aspect.
  *
- * Return value: A #GtkAdjustment object, or %NULL if none is currently being
- * used.
+ * Return value: (transfer none): A #GtkAdjustment object, or %NULL
+ *     if none is currently being used.
  **/
 GtkAdjustment *
 gtk_tree_view_get_vadjustment (GtkTreeView *tree_view)
@@ -11432,8 +11432,8 @@ gtk_tree_view_insert_column_with_data_func  (GtkTreeView               *tree_vie
  *
  * Gets the #GtkTreeViewColumn at the given position in the #tree_view.
  *
- * Return value: The #GtkTreeViewColumn, or %NULL if the position is outside the
- * range of columns.
+ * Return value: (transfer none): The #GtkTreeViewColumn, or %NULL if the
+ *     position is outside the range of columns.
  **/
 GtkTreeViewColumn *
 gtk_tree_view_get_column (GtkTreeView *tree_view,
@@ -11565,10 +11565,10 @@ gtk_tree_view_set_expander_column (GtkTreeView       *tree_view,
  * gtk_tree_view_get_expander_column:
  * @tree_view: A #GtkTreeView
  *
- * Returns the column that is the current expander column.  This
- * column has the expander arrow drawn next to it.
+ * Returns the column that is the current expander column.
+ * This column has the expander arrow drawn next to it.
  *
- * Return value: The expander column.
+ * Return value: (transfer none): The expander column.
  **/
 GtkTreeViewColumn *
 gtk_tree_view_get_expander_column (GtkTreeView *tree_view)
@@ -12406,7 +12406,7 @@ gtk_tree_view_map_expanded_rows_helper (GtkTreeView            *tree_view,
 /**
  * gtk_tree_view_map_expanded_rows:
  * @tree_view: A #GtkTreeView
- * @func: A function to be called
+ * @func: (scope call): A function to be called
  * @data: User data to be passed to the function.
  *
  * Calls @func on all expanded rows.
@@ -12747,12 +12747,13 @@ gtk_tree_view_set_cursor_on_cell (GtkTreeView       *tree_view,
 /**
  * gtk_tree_view_get_bin_window:
  * @tree_view: A #GtkTreeView
- * 
- * Returns the window that @tree_view renders to.  This is used primarily to
- * compare to <literal>event->window</literal> to confirm that the event on
- * @tree_view is on the right window.
- * 
- * Return value: A #GdkWindow, or %NULL when @tree_view hasn't been realized yet
+ *
+ * Returns the window that @tree_view renders to.
+ * This is used primarily to compare to <literal>event->window</literal>
+ * to confirm that the event on @tree_view is on the right window.
+ *
+ * Return value: (transfer none): A #GdkWindow, or %NULL when @tree_view
+ *     hasn't been realized yet
  **/
 GdkWindow *
 gtk_tree_view_get_bin_window (GtkTreeView *tree_view)
@@ -13737,10 +13738,10 @@ gtk_tree_view_get_dest_row_at_pos (GtkTreeView             *tree_view,
  * @tree_view: a #GtkTreeView
  * @path: a #GtkTreePath in @tree_view
  *
- * Creates a #GdkPixmap representation of the row at @path.  
+ * Creates a #GdkPixmap representation of the row at @path.
  * This image is used for a drag icon.
  *
- * Return value: a newly-allocated pixmap of the drag icon.
+ * Return value: (transfer none): a newly-allocated pixmap of the drag icon.
  **/
 GdkPixmap *
 gtk_tree_view_create_row_drag_icon (GtkTreeView  *tree_view,
@@ -14065,7 +14066,7 @@ gtk_tree_view_set_search_equal_func (GtkTreeView                *tree_view,
  * entry for @tree_view.  In case the built-in entry is being used, %NULL
  * will be returned.
  *
- * Return value: the entry currently in use as search entry.
+ * Return value: (transfer none): the entry currently in use as search entry.
  *
  * Since: 2.10
  */
diff --git a/gtk/gtktreeviewcolumn.c b/gtk/gtktreeviewcolumn.c
index aa93ee3..1e646ab 100644
--- a/gtk/gtktreeviewcolumn.c
+++ b/gtk/gtktreeviewcolumn.c
@@ -2271,11 +2271,12 @@ gtk_tree_view_column_set_widget (GtkTreeViewColumn *tree_column,
 /**
  * gtk_tree_view_column_get_widget:
  * @tree_column: A #GtkTreeViewColumn.
- * 
- * Returns the #GtkWidget in the button on the column header.  If a custom
- * widget has not been set then %NULL is returned.
- * 
- * Return value: The #GtkWidget in the column header, or %NULL
+ *
+ * Returns the #GtkWidget in the button on the column header.
+ * If a custom widget has not been set then %NULL is returned.
+ *
+ * Return value: (transfer none): The #GtkWidget in the column
+ *     header, or %NULL
  **/
 GtkWidget *
 gtk_tree_view_column_get_widget (GtkTreeViewColumn *tree_column)
@@ -3767,12 +3768,12 @@ gtk_tree_view_column_queue_resize (GtkTreeViewColumn *tree_column)
  * gtk_tree_view_column_get_tree_view:
  * @tree_column: A #GtkTreeViewColumn
  *
- * Returns the #GtkTreeView wherein @tree_column has been inserted.  If
- * @column is currently not inserted in any tree view, %NULL is
+ * Returns the #GtkTreeView wherein @tree_column has been inserted.
+ * If @column is currently not inserted in any tree view, %NULL is
  * returned.
  *
- * Return value: The tree view wherein @column has been inserted if any,
- *               %NULL otherwise.
+ * Return value: (transfer none): The tree view wherein @column has
+ *     been inserted if any, %NULL otherwise.
  *
  * Since: 2.12
  */
diff --git a/gtk/gtkviewport.c b/gtk/gtkviewport.c
index 933b052..e564b9a 100644
--- a/gtk/gtkviewport.c
+++ b/gtk/gtkviewport.c
@@ -302,10 +302,10 @@ gtk_viewport_destroy (GtkObject *object)
 /**
  * gtk_viewport_get_hadjustment:
  * @viewport: a #GtkViewport.
- * 
+ *
  * Returns the horizontal adjustment of the viewport.
  *
- * Return value: the horizontal adjustment of @viewport.
+ * Return value: (transfer none): the horizontal adjustment of @viewport.
  **/
 GtkAdjustment*
 gtk_viewport_get_hadjustment (GtkViewport *viewport)
@@ -324,7 +324,7 @@ gtk_viewport_get_hadjustment (GtkViewport *viewport)
  * 
  * Returns the vertical adjustment of the viewport.
  *
- * Return value: the vertical adjustment of @viewport.
+ * Return value: (transfer none): the vertical adjustment of @viewport.
  **/
 GtkAdjustment*
 gtk_viewport_get_vadjustment (GtkViewport *viewport)
@@ -581,7 +581,7 @@ gtk_viewport_get_shadow_type (GtkViewport *viewport)
  *
  * Gets the bin window of the #GtkViewport.
  *
- * Return value: a #GdkWindow
+ * Return value: (transfer none): a #GdkWindow
  *
  * Since: 2.20
  **/
diff --git a/gtk/gtkwidget.c b/gtk/gtkwidget.c
index 90d549e..2425392 100644
--- a/gtk/gtkwidget.c
+++ b/gtk/gtkwidget.c
@@ -7297,12 +7297,12 @@ gtk_widget_update_pango_context (GtkWidget *widget)
 /**
  * gtk_widget_create_pango_context:
  * @widget: a #GtkWidget
- * 
+ *
  * Creates a new #PangoContext with the appropriate font map,
  * font description, and base direction for drawing text for
  * this widget. See also gtk_widget_get_pango_context().
- * 
- * Return value: the new #PangoContext
+ *
+ * Return value: (transfer full): the new #PangoContext
  **/
 PangoContext *
 gtk_widget_create_pango_context (GtkWidget *widget)
@@ -7333,7 +7333,7 @@ gtk_widget_create_pango_context (GtkWidget *widget)
  * gtk_widget_create_pango_layout:
  * @widget: a #GtkWidget
  * @text: text to set on the layout (can be %NULL)
- * 
+ *
  * Creates a new #PangoLayout with the appropriate font map,
  * font description, and base direction for drawing text for
  * this widget.
@@ -7341,10 +7341,10 @@ gtk_widget_create_pango_context (GtkWidget *widget)
  * If you keep a #PangoLayout created in this way around, in order to
  * notify the layout of changes to the base direction or font of this
  * widget, you must call pango_layout_context_changed() in response to
- * the #GtkWidget::style-set and #GtkWidget::direction-changed signals 
+ * the #GtkWidget::style-set and #GtkWidget::direction-changed signals
  * for the widget.
- * 
- * Return value: the new #PangoLayout
+ *
+ * Return value: (transfer full): the new #PangoLayout
  **/
 PangoLayout *
 gtk_widget_create_pango_layout (GtkWidget   *widget,
@@ -7385,7 +7385,8 @@ gtk_widget_create_pango_layout (GtkWidget   *widget,
  * the application and should not be modified. The pixbuf should be freed
  * after use with g_object_unref().
  *
- * Return value: a new pixbuf, or %NULL if the stock ID wasn't known
+ * Return value: (transfer full): a new pixbuf, or %NULL if the
+ *     stock ID wasn't known
  **/
 GdkPixbuf*
 gtk_widget_render_icon (GtkWidget      *widget,
diff --git a/gtk/gtkwindow.c b/gtk/gtkwindow.c
index bd12453..b94d089 100644
--- a/gtk/gtkwindow.c
+++ b/gtk/gtkwindow.c
@@ -1711,7 +1711,7 @@ gtk_window_set_default (GtkWindow *window,
  * Returns the default widget for @window. See gtk_window_set_default()
  * for more details.
  *
- * Returns: the default widget, or %NULL if there is none.
+ * Returns: (transfer none): the default widget, or %NULL if there is none.
  *
  * Since: 2.14
  **/
@@ -7520,7 +7520,7 @@ gtk_window_check_screen (GtkWindow *window)
     }
 }
 
-/** 
+/**
  * gtk_window_get_screen:
  * @window: a #GtkWindow.
  *
-- 
1.7.2.3

